--- apps/Makefile.old	2012-01-26 18:19:50.000000000 +0100
+++ apps/Makefile	2012-06-30 14:32:35.000000000 +0200
@@ -18,7 +18,8 @@
 # ----------------------------------------------------------------------------
 
 
-PREFIX = /usr/local
+CXX ?= g++
+PREFIX ?= /usr/local
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 LIBDIR = lib$(SUFFIX)
 DISTDIR = zresample-$(VERSION)
@@ -26,8 +27,8 @@
 MANDIR = /usr/share/man/man1
 LDFLAGS += -L$(PREFIX)/$(LIBDIR)
 CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O3  -ffast-math -Wall
-CXXFLAGS += -march=native
+CXXFLAGS += -Wall
+#CXXFLAGS += -march=native
 
 
 all:	zresample zresample.1.gz
@@ -36,7 +37,7 @@
 ZRESAMPLE_O = zresample.o audiofile.o dither.o
 zresample:	LDLIBS += -lzita-resampler -lsndfile -lrt
 zresample:	$(ZRESAMPLE_O)
-	g++ $(LDFLAGS) -o $@ $(ZRESAMPLE_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ZRESAMPLE_O) $(LDLIBS)
 $(ZRESAMPLE_O):
 -include $(ZRESAMPLE_O:%.o=%.d)
 
