--- GNUmakefile.orig	2007-01-29 01:17:13.000000000 +0100
+++ GNUmakefile	2007-01-29 01:20:26.000000000 +0100
@@ -18,7 +18,7 @@
 #   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-CC = gcc -c -I/usr/include/python2.4 -Wall -Werror $(shell pkg-config --cflags lash-1.0)
+CC = gcc -c -I/usr/include/python2.4 -Wall -Werror -fPIC $(shell pkg-config --cflags lash-1.0)
 
 pylash: _lash.so lash.py
 
