--- ./site_scons/site_init.py.old	2010-01-07 01:57:09.000000000 +0400
+++ ./site_scons/site_init.py	2010-03-13 20:42:21.000000000 +0400
@@ -202,4 +202,5 @@
 
 	def CheckVala(ctx, min_version, fail=True):
+		min_version_breakdown = map(int, re.findall('[0-9]+', min_version))
 		if not SCons.Util.is_String(min_version):
 			raise SCons.Errors.UserError('valac min version needs to be a string')
@@ -209,10 +210,10 @@
 			proc = subprocess.Popen([cmd, '--version'], stdout=subprocess.PIPE)
 			proc.wait()
-			res = re.findall('([0-9](\.[0-9])*)$', proc.stdout.read())
+			res = map(int, re.findall('[0-9]+', proc.stdout.read()))
 		except OSError:
 			ctx.Result(0)
 			raise SCons.Errors.UserError('No vala compiler found')
 
-		if res and res[0] and res[0][0] >= min_version:
+		if res >= min_version_breakdown:
 			ctx.Result(1)
 		else:
