Index: src/Flow/Networks/OSCControlDispatcher.cxx
===================================================================
--- src/Flow/Networks/OSCControlDispatcher.cxx	(revision 9816)
+++ src/Flow/Networks/OSCControlDispatcher.cxx	(working copy)
@@ -41,7 +41,6 @@
 
 		//Init thread
 		mThread.SetThreadCode( makeMemberFunctor0( *mReceiveSocket, UdpListeningReceiveSocket, Run ) );
-		mThread.SetupPriorityPolicy();
 	}
 	OSCControlDispatcher::~OSCControlDispatcher()
 	{
