--- Makefile.orig	2006-05-24 11:30:10.148979361 +0200
+++ Makefile	2006-05-24 11:32:01.574978445 +0200
@@ -16,6 +16,9 @@
 
 
 PREFIX = /usr
+DESTDIR =
+RM = rm
+INSTALL = /usr/bin/install
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 LIBDIR = lib$(SUFFIX)
 VERSION = 0.1.8
@@ -31,23 +34,24 @@
 
 
 japa:	$(JAPA_O)
-	g++ $(LDFLAGS) -o japa $(JAPA_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o japa $(JAPA_O) $(LDLIBS)
 
 
 install:	japa
-	/usr/bin/install -m 755 japa $(PREFIX)/bin
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -m 755 japa $(DESTDIR)$(PREFIX)/bin
 
 $(JAPA_O):	$(JAPA_H)
 
 
 clean:
-	/bin/rm -f *~ *.o *.a *.so japa
+	$(RM) -f *~ *.o *.a *.so japa
 
 
 tarball:
 	cd ..; \
-	/bin/rm -f -r $(DISTDIR)*; \
+	$(RM) -f -r $(DISTDIR)*; \
 	svn export japa $(DISTDIR); \
 	tar cvf $(DISTDIR).tar $(DISTDIR); \
 	bzip2 $(DISTDIR).tar; \
-	/bin/rm -f -r $(DISTDIR);
+	$(RM) -f -r $(DISTDIR);
