diff -burN fst-1.6/fst/Makefile.in fst-1.6-mod/fst/Makefile.in
--- fst-1.6/fst/Makefile.in	2004-04-16 06:41:17.000000000 -0500
+++ fst-1.6-mod/fst/Makefile.in	2004-08-22 16:17:20.631885898 -0500
@@ -95,7 +95,7 @@
 	$(LDSHARED) $(LDDLLFLAGS) -o $@ $(fst_exe_OBJS) $(fst_exe_MODULE).spec.o $(fst_exe_LIBRARY_PATH) $(ALL_LIBRARY_PATH) $(fst_exe_LIBRARIES:%=-l%) $(ALL_LIBRARIES) $(LIBS)
 
 fstconfig: $(fst_exe_MODULE).so fstconfig.o
-	$(CC) -o fstconfig fstconfig.o -L. -lfst 
+	$(CC) -o fstconfig fstconfig.o -L. -lfst $(WINE_LIBRARY_PATH) $(WINE_LIBRARIES)
 
 
 top_distdir = ..
