--- pam-0.99.7.1-r1.ebuild	2007-01-25 18:42:30.000000000 +0100
+++ pam-0.99.7.1-r1.ebuild	2007-02-08 22:07:12.000000000 +0100
@@ -39,13 +38,6 @@
 src_unpack() {
 	unpack ${A}
 	cd "${S}"
-cat << EOT >> "${S}/modules/pam_limits/limits.conf" 
-
-# REALTIME support for audio group users
-@audio          -       rtprio          100
-@audio          -       nice            -10
-@audio          -       memlock         250000
-EOT
 
 	mkdir -p doc/txts
 	for readme in modules/pam_*/README; do
@@ -121,12 +107,4 @@
 	elog "like pam_stack is not present (replaced by the 'include' directive)."
 	elog "The pam_userdb module is now moved in sys-auth/pam_userdb."
 	elog "The pam_console module is now moved in sys-auth/pam_console."
-	elog
-	elog "Please edit file /etc/security/limits.conf"
-	elog "and change this settings to suit your needs"
-	elog
-	elog "@audio          -       rtprio          100"
-	elog "@audio          -       nice            -10"
-	elog "@audio          -       memlock         250000"
-	elog
 }
