From ee9377f283b00e7cfcf2755389202b56bf2bab75 Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Mon, 18 Mar 2013 17:50:31 +0000
Subject: [PATCH 1/2] create prefix dir for installation

---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 53dca65..75f499f 100755
--- a/SConstruct
+++ b/SConstruct
@@ -4,7 +4,7 @@ import glob
 import sys
 
 options = Variables('options.cache', ARGUMENTS)
-options.Add(PathVariable('prefix', 'The prefix where the application will be installed', ''))
+options.Add(PathVariable('prefix', 'The prefix where the application will be installed', '', PathVariable.PathIsDirCreate))
 options.Add(PathVariable('clam_prefix', 'The prefix where CLAM was installed', ''))
 options.Add(BoolVariable('release', 'Enabling compiler optimizations', 'no') )
 options.Add(BoolVariable('verbose', 'Display the full command line instead a short command description', 'no') )
-- 
1.7.12.4

