--- Makefile.orig	2007-01-30 20:31:31.633172665 +0100
+++ Makefile	2007-01-30 21:42:38.223035750 +0100
@@ -33,7 +33,7 @@
 ## standard
 #PDEXECUTABLE = /usr/local/bin/pd
 # standard dev layout location
-PDEXECUTABLE = ../../../pd/bin/pd
+PDEXECUTABLE = /usr/bin/pd
 
 
 # generic optimization
@@ -49,9 +49,9 @@
 # faster G4 7450 optimization on Mac OS X ( -fast gives errors, use -O3 instead)
 #OPT_FLAGS = -O3 -mcpu=7450 -faltivec -ffast-math -fPIC
 
-CFLAGS = $(OPT_FLAGS) -Wall -W -Wno-shadow -Wstrict-prototypes -Wno-unused
+CFLAGS = $(OUR_CFLAGS) -fPIC
 
-INCLUDE =  -I./ -I../../../pd/src -I./HID\ Utilities\ Source
+INCLUDE =  -I./ -I/usr/include -I./HID\ Utilities\ Source
 
 .c.o:
 	$(CC) $(CFLAGS) $(INCLUDE) -c *.c
