--- guitarix-0.25.2.orig/wscript	2013-01-29 14:36:39.000000000 +0100
+++ guitarix-0.25.2.a/wscript	2013-03-30 18:09:17.965348694 +0100
@@ -753,11 +753,6 @@
         task.scan = po_scanner
         return task
     gen_desktop.create_task = gen_desktop_add_scanner
-    if Options.commands["install"]:  ## newer waf version: bld.cmd == 'install'
-        try:
-            bld.exec_command(["update-desktop-database", Utils.subst_vars("${DATADIR}/applications", bld.env)])
-        except:
-            pass
 
     if bld.env["GX_LIB_SHARED"]:
         bld.add_post_fun(post)
