diff -u guitarix2-0.18.0-unpatched/wscript guitarix2-0.18.0-patched/wscript
--- guitarix2-0.18.0-unpatched/wscript	2011-08-09 18:10:07.000000000 +0300
+++ guitarix2-0.18.0-patched/wscript	2011-08-10 23:52:04.714381135 +0300
@@ -72,8 +72,6 @@
     group = opt.add_option_group ('Localization and documentation', '')
     add_enable_option ('nls', 'native language support', group)
 
-    if 'LINGUAS' in os.environ: conf.env['LINGUAS'] = os.environ['LINGUAS']
-    
     opt.tool_options('compiler_cxx')
     comp = opt.get_option_group("--check-cxx-compiler")
 
@@ -382,6 +380,8 @@
 
 # guitarix waf configuration
 def configure(conf):
+    if 'LINGUAS' in os.environ: conf.env['LINGUAS'] = os.environ['LINGUAS']
+
     platform = Utils.detect_platform()
     conf.env['IS_MACOSX'] = platform == 'darwin'
     conf.env['IS_LINUX'] = platform == 'linux' or platform == 'posix'
