diff -ur screencastor_test.orig/go_screencastor.sh screencastor_test/go_screencastor.sh
--- screencastor_test.orig/go_screencastor.sh	2013-06-09 18:01:50.000000000 +0200
+++ screencastor_test/go_screencastor.sh	2013-08-11 15:08:52.000000000 +0200
@@ -17,20 +17,20 @@
 	do
 		case "${option}" in
 			"--help")
-				echo -e "${BLEUFONCE}$(eval_gettext $'For more information:') ${FUSHIA}man screencastor${RAZ}" ;;
+				echo -e "${BLEUFONCE}$(gettext $'For more information:') ${FUSHIA}man screencastor${RAZ}" ;;
 
 			"--pause")
-				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(eval_gettext $'Pause of the screencast')${RAZ}"; touch "${tmp}/pause"; } ;;
+				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(gettext $'Pause of the screencast')${RAZ}"; touch "${tmp}/pause"; } ;;
 
 			"--stop")
-				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(eval_gettext $'Stop of the screencast')${RAZ}"; touch "${tmp}/stop"; } ;;
+				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(gettext $'Stop of the screencast')${RAZ}"; touch "${tmp}/stop"; } ;;
 
 			"--start")
-				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(eval_gettext $'Start of the screencast')${RAZ}"; touch "${tmp}/start"; } ;;
+				[[ ${tmp} ]] && { echo -e "${BLEUFONCE}$(gettext $'Start of the screencast')${RAZ}"; touch "${tmp}/start"; } ;;
 
 			"--version")
-				echo -e "${BLEUFONCE}$(eval_gettext $'Screencastor version:') ${FUSHIA}$(sed -n "s/^version=.\(.*\)./\1/p" "screencastor.sh")${RAZ}"
-				echo -e "${BLEUFONCE}$(eval_gettext $'Screencastor last modification:') ${FUSHIA}$(stat -c %y "screencastor.sh" | cut -d" " -f 1)${RAZ}" ;;
+				echo -e "${BLEUFONCE}$(gettext $'Screencastor version:') ${FUSHIA}$(sed -n "s/^version=.\(.*\)./\1/p" "screencastor.sh")${RAZ}"
+				echo -e "${BLEUFONCE}$(gettext $'Screencastor last modification:') ${FUSHIA}$(stat -c %y "screencastor.sh" | cut -d" " -f 1)${RAZ}" ;;
 		esac
 	done
 
diff -ur screencastor_test.orig/screencastor.py screencastor_test/screencastor.py
--- screencastor_test.orig/screencastor.py	2013-06-09 18:01:49.000000000 +0200
+++ screencastor_test/screencastor.py	2013-08-11 12:31:59.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding:Utf­8 ­-*-
 ############################################################################
 ##                                                                        ##
