--- layout.c.orig	2007-03-15 21:54:51.000000000 +0600
+++ layout.c	2007-03-15 21:53:51.000000000 +0600
@@ -326,6 +326,7 @@
 	grow_cmp = new_grow < old_grow;
 
 	/* XXX: optimize in one expression? */
+	if (!old->valid && new->valid) goto replace;
 	if (settings.min_space_policy && grow_cmp) goto replace;
 	if (old_grow && !new_grow) goto replace;
 	if ((!old_grow == !new_grow) && compare_points(new->pos, old->pos)) goto replace;
