--- aliki/Makefile.old	2010-04-12 14:40:41.000000000 +0100
+++ aliki/Makefile	2010-04-12 14:53:03.000000000 +0100
@@ -22,9 +22,10 @@
 # Modify as required.
 #
 #PREFIX = /usr/local
-PREFIX = /usr
+CXX ?= g++
+PREFIX ?= /usr
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-LIBDIR = lib$(SUFFIX)
+LIBDIR ?= lib$(SUFFIX)
 VERSION = 0.0.3
 DISTDIR = aliki-$(VERSION)
 CPPFLAGS += -O3 -Wall -MMD -MP -DVERSION=\"$(VERSION)\" -DPREFIX=\"$(PREFIX)\" -I/usr/include/freetype2
@@ -44,7 +45,7 @@
 aliki:	LDLIBS += -lclxclient -lclthreads -lpthread -lsndfile -lfftw3f -lXft -lX11 -lrt
 aliki:	LDFLAGS += -L$(PREFIX)/$(LIBDIR) -L/usr/X11R6/$(LIBDIR)
 aliki:	$(ALIKI_O)
-	g++ $(LDFLAGS) -o $@ $(ALIKI_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ALIKI_O) $(LDLIBS)
 $(ALIKI_O):
 -include $(ALIKI_O:%.o=%.d)
 
@@ -54,15 +55,15 @@
 aliki-rt:	LDLIBS += -lclalsadrv -lclthreads -lsndfile -ljack -lasound -lpthread -lrt
 aliki-rt:	LDFLAGS += -L$(PREFIX)/$(LIBDIR)
 aliki-rt:	 $(ALIKI_RT_O) $(LIBCLX)
-	g++ $(LDFLAGS) -o $@ $(ALIKI_RT_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ALIKI_RT_O) $(LDLIBS)
 $(ALIKI_RT_O):
 -include $(ALIKI_RT_O:%.o=%.d)
 
 
 
 install:	aliki aliki-rt
-	/usr/bin/install -m 755 aliki    $(PREFIX)/bin
-	/usr/bin/install -m 755 aliki-rt $(PREFIX)/bin
+	/usr/bin/install -D -m 755 aliki    $(DESTDIR)$(PREFIX)/bin/aliki
+	/usr/bin/install -D -m 755 aliki-rt $(DESTDIR)$(PREFIX)/bin/aliki-rt
 
 clean:
 	/bin/rm -f *~ *.o *.a *.d *.so
