--- Makefile.orig	2006-05-24 11:10:36.108989019 +0200
+++ Makefile	2006-05-24 11:13:03.598987806 +0200
@@ -16,6 +16,9 @@
 
 
 PREFIX = /usr
+DESTDIR =
+INSTALL = /usr/bin/install
+RM = rm
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 LIBDIR = lib$(SUFFIX)
 VERSION = 0.4.0
@@ -31,23 +34,24 @@
 
 
 jaaa:	$(JAAA_O)
-	g++ $(LDFLAGS) -o jaaa $(JAAA_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o jaaa $(JAAA_O) $(LDLIBS)
 
 
 install:	jaaa
-	/usr/bin/install -m 755 jaaa $(PREFIX)/bin
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -m 755 jaaa $(DESTDIR)$(PREFIX)/bin
 
 $(JAAA_O):	$(JAAA_H)
 
 
 clean:
-	/bin/rm -f *~ *.o *.a *.so jaaa
+	$(RM) -f *~ *.o *.a *.so jaaa
 
 
 tarball:
 	cd ..; \
-	/bin/rm -f -r $(DISTDIR)*; \
+	$(RM) -f -r $(DISTDIR)*; \
 	svn export jaaa $(DISTDIR); \
 	tar cvf $(DISTDIR).tar $(DISTDIR); \
 	bzip2 $(DISTDIR).tar; \
-	/bin/rm -f -r $(DISTDIR);
+	$(RM) -f -r $(DISTDIR);
