--- lilypond-2.10.1/configure	2006-12-03 17:38:49.000000000 +0100
+++ lilypond-2.10.0/configure	2006-11-10 14:44:30.000000000 +0100
@@ -1670,7 +1670,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -1813,7 +1813,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -1957,7 +1957,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -2216,7 +2216,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -3594,7 +3594,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -3802,7 +3802,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -3931,7 +3931,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -4409,7 +4409,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -4703,7 +4703,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -5177,7 +5177,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -5254,7 +5254,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -5405,7 +5405,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -6295,7 +6295,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -6630,7 +6630,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -6775,7 +6775,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -6938,7 +6938,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -7311,7 +7311,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -7841,7 +7841,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -8230,7 +8230,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -8375,7 +8375,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -8558,7 +8558,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -9748,7 +9748,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -10683,7 +10683,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -10873,7 +10873,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -11061,7 +11061,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
@@ -11211,7 +11211,7 @@
     ## grab the first version number in  --version output.
     eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '\(^\| \)[0-9][0-9]*\.[0-9]' \
         | head -n 1 \
-	| tr ' ' '\n' | sed 's/\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' | grep '\(^\| \)[0-9][0-9]*\.[0-9]' | head -n 1\`\"
+	| tr ' ' '\n' | grep '[0-9]\.[0-9]' | head -n 1 | sed 's/\([0-9.]*\).*/\1/g'\`\"
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
