--- zita-convolver-2.0.0/libs/Makefile.old	2011-01-19 08:14:23.574717257 +0000
+++ zita-convolver-2.0.0/libs/Makefile	2011-01-19 08:21:32.962716987 +0000
@@ -17,9 +17,9 @@
 
 # Modify as required.
 #
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-LIBDIR = lib$(SUFFIX)
+LIBDIR ?= lib$(SUFFIX)
 
 
 MAJVERS = 2
@@ -28,9 +28,9 @@
 DISTDIR = zita-convolver-$(VERSION)
 
 
-CPPFLAGS += -Wall -I. -fPIC -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -O3
-CPPFLAGS += -mmmx -msse -mfpmath=sse -ffast-math -funroll-loops -fpermissive
-CPPFLAGS += -march=i686
+CPPFLAGS += -Wall -I. -fPIC -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
+CPPFLAGS += -ffast-math -funroll-loops -fpermissive
+#CPPFLAGS += -march=i686
 #CPPFLAGS += -DVECTORIZE 
 
 LDLFAGS += 
@@ -46,16 +46,16 @@
 
 
 $(ZITA-CONVOLVER_MIN):	$(ZITA-CONVOLVER_O)
-	g++ -shared $(LDFLAGS) -Wl,-soname,$(ZITA-CONVOLVER_MAJ) -o $(ZITA-CONVOLVER_MIN) $(ZITA-CONVOLVER_O) $(ZITA-CONVOLVER_DEP)
+	$(CXX) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-CONVOLVER_MAJ) -o $(ZITA-CONVOLVER_MIN) $(ZITA-CONVOLVER_O) $(ZITA-CONVOLVER_DEP)
 
 
 install:	$(ZITA-CONVOLVER_MIN)
-	install -d $(PREFIX)/include
-	install -d $(PREFIX)/$(LIBDIR)
-	install -m 644 $(ZITA-CONVOLVER_H) $(PREFIX)/include
-	install -m 755 $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_SO)
+	install -d $(DESTDIR)$(PREFIX)/include
+	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -m 644 $(ZITA-CONVOLVER_H) $(DESTDIR)$(PREFIX)/include
+	install -m 755 $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	#ldconfig
+	ln -sf $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_SO)
 
 
 clean:
