*** Makefile.orig	2007-12-19 12:46:08.000000000 +0100
--- Makefile	2008-01-07 22:57:33.000000000 +0100
***************
*** 46,57 ****
  
  
  install:	$(ZITA-CONVOLVER_MIN)
! 	/usr/bin/install -d $(PREFIX)/include
! 	/usr/bin/install -d $(PREFIX)/$(LIBDIR)
! 	/usr/bin/install -m 644 $(ZITA-CONVOLVER_H) $(PREFIX)/include
! 	/usr/bin/install -m 755 $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)
! 	/sbin/ldconfig -n $(PREFIX)/$(LIBDIR)
! 	ln -sf $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_SO)
  
  
  clean:
--- 46,57 ----
  
  
  install:	$(ZITA-CONVOLVER_MIN)
! 	install -d $(DESTDIR)$(PREFIX)/include $(DESTDIR)$(PREFIX)/lib
! 	install -d $(PREFIX)/$(LIBDIR)
! 	install -m 644 $(ZITA-CONVOLVER_H) $(DESTDIR)$(PREFIX)/include
! 	install -m 755 $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/lib
! 	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
! 	ln -sf $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/lib/$(ZITA-CONVOLVER_SO)
  
  
  clean:
