diff -ruN rosegarden-4-1.0.orig/base/Colour.C rosegarden-4-1.0/base/Colour.C
--- rosegarden-4-1.0.orig/base/Colour.C	2005-02-08 20:55:22.000000000 +0600
+++ rosegarden-4-1.0/base/Colour.C	2005-12-30 14:24:19.000000000 +0600
@@ -163,7 +163,7 @@
 // Generic Colour routines:
 
 Colour
-Rosegarden::getCombinationColour(const Colour &input1, const Colour &input2)
+getCombinationColour(const Colour &input1, const Colour &input2)
 {
     Colour ret((input1.getRed()+input2.getRed())/2,
                 (input1.getGreen()+input2.getGreen())/2,
