--- Makefile.in.orig	2007-05-20 18:31:40.000000000 +0200
+++ Makefile.in	2007-05-20 18:32:25.000000000 +0200
@@ -640,11 +640,6 @@
 	    fi \
 	  done \
 	fi
-	if test -f $(srcdir)/phasex.desktop; then \
-	  $(INSTALL_DATA) $(srcdir)/phasex.desktop $(DESTDIR)$(pkgdatadir); \
-	  desktop-file-install --vendor=sysex --add-category="Audio" --add-category="AudioVideo" \
-	    --add-category="X-MIDI" --add-category="X-JACK" $(srcdir)/phasex.desktop; \
-	fi
 
 dist-hook:
 	if test -d misc; then \
