--- jack_oscrolloscope-0.6/Makefile.old	2010-11-09 18:24:34.000000000 +0000
+++ jack_oscrolloscope-0.6/Makefile	2010-11-09 18:36:01.000000000 +0000
@@ -1,9 +1,10 @@
-PREFIX =	/usr/local
+CC ?= gcc
+PREFIX ?=	/usr/local
 
 CFLAGS +=	$(shell sdl-config --cflags) $(shell pkg-config --cflags jack) -Wall -std=gnu99
 LIBS =		$(shell sdl-config --libs) $(shell pkg-config --libs jack) -lGL
 
-CFLAGS +=	-O2
+#CFLAGS +=	-O2
 
 OBJS =		main.o video.o audio.o waves.o
 BIN =		jack_oscrolloscope
@@ -12,16 +13,16 @@
 all:		$(BIN)
 
 $(BIN): $(OBJS)
-		gcc -o $(BIN) $(OBJS) $(LIBS)
+		$(CC) $(LDFLAGS) -o $(BIN) $(OBJS) $(LIBS)
 
 %.o: %.c
-		gcc -c -o $@ $(CFLAGS) $*.c
+		$(CC) -c -o $@ $(CFLAGS) $*.c
 
 clean:
 		rm -f $(BIN) $(OBJS) .dep
 
 install: $(BIN)
-	/usr/bin/install -m 755 $(BIN) $(PREFIX)/bin
+	/usr/bin/install -Dm 755 $(BIN) $(DESTDIR)$(PREFIX)/bin/$(BIN)
 
 
 dep:
