--- a/ardour-2.0.2/gtk2_ardour/SConscript	2007-05-07 06:38:55.000000000 +0200
+++ b/ardour-2.0.2/gtk2_ardour/SConscript	2007-06-10 16:50:04.000000000 +0200
@@ -35,7 +35,7 @@
     libraries['libgnomecanvas2'],
     libraries['libgnomecanvasmm'],
     libraries['sysmidi'],
-    libraries['sndfile-ardour'],
+    libraries['sndfile'],
     libraries['flac'],
     libraries['lrdf'],
     libraries['glibmm2'],
--- a/ardour-2.0.2/libs/ardour/SConscript	2007-05-07 06:39:05.000000000 +0200
+++ b/ardour-2.0.2/libs/ardour/SConscript	2007-06-10 16:50:17.000000000 +0200
@@ -254,7 +254,7 @@
 ardour.Merge ([
 	     libraries['core'],
 	     libraries['xml'],
-	     libraries['sndfile-ardour'],
+	     libraries['sndfile'],
 	     libraries['raptor'],
 	     libraries['lrdf'],
 	     libraries['samplerate'],
--- a/ardour-2.0.2/libs/surfaces/control_protocol/SConscript	2007-05-07 06:39:24.000000000 +0200
+++ b/ardour-2.0.2/libs/surfaces/control_protocol/SConscript	2007-06-10 16:50:40.000000000 +0200
@@ -33,7 +33,7 @@
 
 cp.Merge ([
     libraries['ardour'],
-    libraries['sndfile-ardour'],
+    libraries['sndfile'],
     libraries['sigc2'],
     libraries['pbd'],
     libraries['midi++2'],
--- a/ardour-2.0.2/libs/surfaces/generic_midi/SConscript	2007-05-07 06:39:25.000000000 +0200
+++ b/ardour-2.0.2/libs/surfaces/generic_midi/SConscript	2007-06-10 16:50:40.000000000 +0200
@@ -34,7 +34,7 @@
 genericmidi.Merge ([
 	libraries['ardour'],
 	libraries['ardour_cp'],
-	libraries['sndfile-ardour'],
+	libraries['sndfile'],
 	libraries['midi++2'],
 	libraries['pbd'],
 	libraries['sigc2'],
--- a/ardour-2.0.2/libs/surfaces/mackie/SConscript	2007-05-07 06:39:25.000000000 +0200
+++ b/ardour-2.0.2/libs/surfaces/mackie/SConscript	2007-06-10 16:50:40.000000000 +0200
@@ -51,7 +51,7 @@
     libraries['xml'],
     libraries['glib2'],
     libraries['glibmm2'],
-    libraries['sndfile-ardour']
+    libraries['sndfile']
     ])
 
 libardour_mackie = mackie.SharedLibrary('ardour_mackie', mackie_files)
--- a/ardour-2.0.2/libs/surfaces/tranzport/SConscript	2007-05-07 06:39:24.000000000 +0200
+++ b/ardour-2.0.2/libs/surfaces/tranzport/SConscript	2007-06-10 16:50:40.000000000 +0200
@@ -63,7 +63,7 @@
     libraries['glib2'],
     libraries['glibmm2'],
     libraries['usb'],
-    libraries['sndfile-ardour']
+    libraries['sndfile']
     ])
 
 libardour_tranzport = tranzport.SharedLibrary('ardour_tranzport', tranzport_files)
--- a/ardour-2.0.2/SConstruct	2007-06-10 17:17:03.000000000 +0200
+++ b/ardour-2.0.2/SConstruct	2007-06-10 17:06:03.000000000 +0200
@@ -819,6 +819,7 @@
 
     syslibdeps = \
     {
+        'sndfile'				  : '1.0.16',
         'sigc++-2.0'           : '2.0',
         'gtkmm-2.4'            : '2.8',
         'libgnomecanvasmm-2.6' : '2.12.0'
@@ -851,14 +852,16 @@
     libraries['pangomm'].ParseConfig ('pkg-config --cflags --libs pangomm-1.4')
     libraries['libgnomecanvasmm'] = LibraryInfo()
     libraries['libgnomecanvasmm'].ParseConfig ('pkg-config --cflags --libs libgnomecanvasmm-2.6')
+    libraries['sndfile'] = LibraryInfo()
+    libraries['sndfile'].ParseConfig('pkg-config --cflags --libs sndfile')
 
 #
 # cannot use system one for the time being
 #
     
-    libraries['sndfile-ardour'] = LibraryInfo(LIBS='libsndfile-ardour',
-                                    LIBPATH='#libs/libsndfile',
-                                    CPPPATH=['#libs/libsndfile/src'])
+#    libraries['sndfile-ardour'] = LibraryInfo(LIBS='libsndfile-ardour',
+#                                    LIBPATH='#libs/libsndfile',
+#                                    CPPPATH=['#libs/libsndfile/src'])
 
 #    libraries['libglademm'] = LibraryInfo()
 #    libraries['libglademm'].ParseConfig ('pkg-config --cflags --libs libglademm-2.4')
@@ -878,7 +881,6 @@
     ]
     
     subdirs = [
-        'libs/libsndfile',
         'libs/pbd',
         'libs/midi++2',
         'libs/ardour',
@@ -926,9 +928,9 @@
     libraries['soundtouch'] = LibraryInfo(LIBS='soundtouch',
                                           LIBPATH='#libs/soundtouch',
                                           CPPPATH=['#libs', '#libs/soundtouch'])
-    libraries['sndfile-ardour'] = LibraryInfo(LIBS='libsndfile-ardour',
-                                    LIBPATH='#libs/libsndfile',
-                                    CPPPATH=['#libs/libsndfile', '#libs/libsndfile/src'])
+#    libraries['sndfile-ardour'] = LibraryInfo(LIBS='libsndfile-ardour',
+#                                    LIBPATH='#libs/libsndfile',
+#                                    CPPPATH=['#libs/libsndfile', '#libs/libsndfile/src'])
 #    libraries['libglademm'] = LibraryInfo(LIBS='libglademm',
 #                                          LIBPATH='#libs/libglademm',
 #                                          CPPPATH='#libs/libglademm')
@@ -943,7 +945,6 @@
     
     subdirs = [
         'libs/sigc++2',
-        'libs/libsndfile',
         'libs/pbd',
         'libs/midi++2',
         'libs/ardour',
--- a/ardour-2.0.2/vst/SConscript	2007-05-07 06:39:37.000000000 +0200
+++ b/ardour-2.0.2/vst/SConscript	2007-06-10 16:50:40.000000000 +0200
@@ -35,7 +35,7 @@
     libraries['libgnomecanvas2'],
     libraries['libgnomecanvasmm'],
     libraries['sysmidi'],
-    libraries['sndfile-ardour'],
+    libraries['sndfile'],
     libraries['flac'],
     libraries['lrdf'],
     libraries['glibmm2'],
