--- src/Makefile.in~	2008-05-21 21:05:58.000000000 +0200
+++ src/Makefile.in	2008-09-18 04:05:11.000000000 +0200
@@ -130,7 +130,7 @@
 CXXCOMPILE = $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
 CXXLD = $(CXX)
-CXXLINK = $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) $(LDFLAGS) \
+CXXLINK = $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
 	-o $@
 SOURCES = $(openmovieeditor_SOURCES)
 DIST_SOURCES = $(openmovieeditor_SOURCES)
@@ -615,7 +615,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 openmovieeditor$(EXEEXT): $(openmovieeditor_OBJECTS) $(openmovieeditor_DEPENDENCIES) 
 	@rm -f openmovieeditor$(EXEEXT)
-	$(CXXLINK) $(openmovieeditor_OBJECTS) $(openmovieeditor_LDADD) $(LIBS)
+	$(CXXLINK) $(openmovieeditor_OBJECTS) $(openmovieeditor_LDADD) $(LIBS) $(LDFLAGS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
