--- seq24-0.8.7/src/mainwid.cpp	2006-05-17 21:31:23.000000000 -0400
+++ seq24-0.8.7-patched/src/mainwid.cpp	2006-11-04 14:11:34.000000000 -0500
@@ -212,7 +212,7 @@ mainwid::draw_sequence_on_pixmap( int a_
 	    char str[20];
 	    sprintf( str, 
 		     "%d-%d %ld/%ld",
-		     seq->get_midi_bus(), 
+		     seq->get_midi_bus()+1, 
 		     seq->get_midi_channel()+1,
 		     seq->get_bpm(), seq->get_bw() );
         