--- seq24-0.8.7/src/seqmenu.cpp	2006-05-30 09:12:05.000000000 -0400
+++ seq24-0.8.7-patched/src/seqmenu.cpp	2006-11-04 14:04:09.000000000 -0500
@@ -79,35 +79,38 @@ seqmenu::popup_menu( void )
         m_menu->items().push_back(SeparatorElem());
         Menu *menu_buses = manage( new Menu() );
 
-        m_menu->items().push_back( MenuElem( "Midi Bus", *menu_buses) );
+        m_menu->items().push_back( MenuElem( "MIDI Bus", *menu_buses) );
 
-        /* midi buses */
+        /* midi bus menu */
         mastermidibus *masterbus = m_mainperf->get_master_midi_bus();
         for ( int i=0; i< masterbus->get_num_out_buses(); i++ ){
+            menu_buses->items().push_back(MenuElem( masterbus->get_midi_out_bus_name(i),
+                    SigC::bind(mem_fun(*this,&seqmenu::set_bus_and_midi_channel), i, -1 )));
+        }
 
-            Menu *menu_channels = manage( new Menu() );
+        char b[4];
+        Menu *menu_channels = manage( new Menu() );
 
-            menu_buses->items().push_back(MenuElem( masterbus->get_midi_out_bus_name(i),
-                        *menu_channels ));
-            char b[4];
+        m_menu->items().push_back( MenuElem( "MIDI Channel", *menu_channels) );
 
-            /* midi channel menu */
-            for( int j=0; j<16; j++ ){
-                sprintf( b, "%d", j+1 );
-                std::string name = string(b);
-                int instrument = global_user_midi_bus_definitions[i].instrument[j]; 
-                if ( instrument >= 0 && instrument < c_maxBuses )
-                {
-                    name = name + (string(" (") + 
-                            global_user_instrument_definitions[instrument].instrument + 
-                            string(")") );
-                }
-
-                menu_channels->items().push_back(MenuElem(name, 
-                            SigC::bind(mem_fun(*this,&seqmenu::set_bus_and_midi_channel), 
-                                i, j )));
+        /* midi channel menu */
+        for( int j=0; j<16; j++ ){
+            sprintf( b, "%d", j+1 );
+            std::string name = string(b);
+            int a_bus = m_mainperf->get_sequence( m_current_seq )->get_midi_bus();
+            int instrument = global_user_midi_bus_definitions[a_bus].instrument[j]; 
+            if ( instrument >= 0 && instrument < c_maxBuses )
+            {
+                name = name + (string(" (") + 
+                        global_user_instrument_definitions[instrument].instrument + 
+                        string(")") );
             }
-        }        
+
+            menu_channels->items().push_back(MenuElem(name, 
+                        SigC::bind(mem_fun(*this,&seqmenu::set_bus_and_midi_channel), 
+                            -1, j )));
+        }
+                
     }
 
     m_menu->popup(0,0);
@@ -118,8 +121,12 @@ seqmenu::popup_menu( void )
 seqmenu::set_bus_and_midi_channel( int a_bus, int a_ch )
 {
     if ( m_mainperf->is_active( m_current_seq )) {
-        m_mainperf->get_sequence( m_current_seq )->set_midi_bus( a_bus );
-        m_mainperf->get_sequence( m_current_seq )->set_midi_channel( a_ch );
+        if ( a_bus >= 0 ) { 
+            m_mainperf->get_sequence( m_current_seq )->set_midi_bus( a_bus );
+        }
+        if ( a_ch >= 0 ) {
+            m_mainperf->get_sequence( m_current_seq )->set_midi_channel( a_ch );
+        }
         m_mainperf->get_sequence( m_current_seq )->set_dirty();
     }
 }