diff -Naur wine-0.9.35.orig/dlls/winealsa.drv/midi.c wine-0.9.35/dlls/winealsa.drv/midi.c
--- wine-0.9.35.orig/dlls/winealsa.drv/midi.c	2007-04-18 19:19:46.000000000 +0200
+++ wine-0.9.35/dlls/winealsa.drv/midi.c	2007-04-18 19:21:48.000000000 +0200
@@ -549,6 +549,7 @@
     if (!(lpMidiHdr->dwFlags & MHDR_PREPARED)) return MIDIERR_UNPREPARED;
 
     EnterCriticalSection(&crit_sect);
+    lpMidiHdr->dwFlags |= MHDR_INQUEUE;
     if (MidiInDev[wDevID].lpQueueHdr == 0) {
 	MidiInDev[wDevID].lpQueueHdr = lpMidiHdr;
     } else {
