--- Makefile	2006-11-23 18:01:20.000000000 +0000
+++ /root/libclthreads-2.2.1-makefile	2006-12-16 12:21:15.000000000 +0000
@@ -46,12 +46,11 @@
 
 
 install:	$(CLTHREADS_MIN)
-	/usr/bin/install -d $(PREFIX)/$(LIBDIR)
-	/usr/bin/install -m 644 $(CLTHREADS_H) $(PREFIX)/include
-	/usr/bin/install -m 755 $(CLTHREADS_MIN) $(PREFIX)/$(LIBDIR)
-	/sbin/ldconfig -n $(PREFIX)/$(LIBDIR)
-	ln -sf $(CLTHREADS_MIN) $(PREFIX)/$(LIBDIR)/$(CLTHREADS_SO)
-
+	install -d $(DESTDIR)$(PREFIX)/include $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -m 644 $(CLTHREADS_H) $(DESTDIR)$(PREFIX)/include
+	install -m 755 $(CLTHREADS_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	ln -sf $(CLTHREADS_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(CLTHREADS_SO)
 
 clean:
 	/bin/rm -f *~ *.o *.a *.d *.so.*
