From 840bf3229de337b3a54c1f7f2b99156c54d64ef2 Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Thu, 4 Apr 2013 04:08:45 +0100
Subject: [PATCH 1/2] printf include

---
 src/sid_chips.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/sid_chips.cc b/src/sid_chips.cc
index d130ebf..bbea1eb 100644
--- a/src/sid_chips.cc
+++ b/src/sid_chips.cc
@@ -5,6 +5,7 @@ using namespace std;
 #include <math.h>
 #include <alloca.h>
 #include <cstdlib>
+#include <stdio.h>
 
 #include "sid_instr.h"
 #include "midi.h"
-- 
1.8.1.5

