--- SConstruct.orig	2010-01-16 03:27:37.000000000 +0400
+++ SConstruct	2010-03-07 17:04:27.000000000 +0400
@@ -505,7 +505,6 @@
 
 if getPlatform() == 'linux':
     commonEnvironment.Append(CCFLAGS = "-DLINUX")
-    commonEnvironment.Append(CPPFLAGS = ['-DHAVE_SOCKETS'])
     commonEnvironment.Append(CPPPATH = '/usr/local/include')
     commonEnvironment.Append(CPPPATH = '/usr/include')
     commonEnvironment.Append(CPPPATH = '/usr/include')
@@ -837,12 +836,6 @@
     if configure.CheckHeader(h[0], language = "C"):
         commonEnvironment.Append(CPPFLAGS = [h[1]])
 
-if getPlatform() == 'win32':
-    if configure.CheckHeader("winsock.h", language = "C"):
-        commonEnvironment.Append(CPPFLAGS = ['-DHAVE_SOCKETS'])
-elif configure.CheckHeader("sys/socket.h", language = "C"):
-    commonEnvironment.Append(CPPFLAGS = ['-DHAVE_SOCKETS'])
-
 if getPlatform() == 'darwin':
     commonEnvironment.Append(CPPFLAGS = ['-DHAVE_DIRENT_H'])
 elif configure.CheckHeader("dirent.h", language = "C"):
