--- SConstruct.org	2007-03-09 17:57:40.000000000 +0100
+++ SConstruct	2007-03-09 17:57:31.000000000 +0100
@@ -1062,7 +1062,7 @@
         print 'CONFIGURATION DECISION: Building Csound Lua interface library.'
         csoundLuaInterface = csoundWrapperEnvironment.SharedObject(
             'interfaces/lua_interface.i',
-            SWIGFLAGS = [swigflags, '-lua', '-outdir', '.'])
+            SWIGFLAGS = [swigflags, '-Ilua', '-outdir', '.'])
         if getPlatform() != 'darwin':
             csoundInterfacesSources.insert(0, csoundLuaInterface)
             if getPlatform() == 'win32':
@@ -1671,7 +1671,7 @@
     vstEnvironment.Append(LIBS = libCsoundLibs)
     vstEnvironment.Append(SWIGFLAGS = Split('-c++ -includeall -verbose -outdir .'))
     if getPlatform() == 'linux':
-        vstEnvironment.Append(LIBS = ['util', 'dl', 'm'])
+        vstEnvironment.Append(LIBS = ['util', 'dl', 'm', 'tcl8.4'])
         vstEnvironment.Append(SHLINKFLAGS = '--no-export-all-symbols')
         vstEnvironment.Append(LINKFLAGS = ['-Wl,-rpath-link,.'])
         guiProgramEnvironment.Prepend(LINKFLAGS = ['-Wl,-rpath-link,.'])
