--- H/remote.h.orig	2007-07-02 13:43:41.000000000 +0200
+++ H/remote.h	2007-07-02 13:44:11.000000000 +0200
@@ -30,6 +30,7 @@
 #else
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <net/if.h>
 #include <arpa/inet.h>
 #ifdef HAVE_UNISTD_H
 #  include <unistd.h>
--- OOps/remote.c.orig	2007-07-02 13:44:48.000000000 +0200
+++ OOps/remote.c	2007-07-02 13:45:31.000000000 +0200
@@ -28,10 +28,9 @@
 #ifndef WIN32
 #include <sys/ioctl.h>
 #ifdef LINUX
-#include <linux/if.h>
+#include <net/if.h>
 #include <arpa/inet.h>
 #endif
-#include <net/if.h>
 #endif
 #endif /* HAVE_SOCKETS */
 
