--- ambdec-0.4.2/source/Makefile.old	2010-04-20 17:26:16.000000000 +0100
+++ ambdec-0.4.2/source/Makefile	2010-04-20 17:28:52.000000000 +0100
@@ -14,10 +14,10 @@
 #  along with this program; if not, write to the Free Software
 #  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-
-PREFIX = /usr/local
+CXX ?= g++
+PREFIX ?= /usr/local
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-LIBDIR = lib$(SUFFIX)
+LIBDIR ?= lib$(SUFFIX)
 SHARED = $(PREFIX)/share
 VERSION = 0.4.2
 DISTDIR = ambdec-$(VERSION)
@@ -36,7 +36,7 @@
 ambdec:	LDLIBS += -lclxclient -lclthreads -ljack -lpng -lXft -lX11 -lrt
 ambdec:	LDFLAGS += -L/usr/X11R6/lib
 ambdec:	$(AMBDEC_O)
-	g++ $(LDFLAGS) -o $@ $(AMBDEC_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(AMBDEC_O) $(LDLIBS)
 
 $(AMBDEC_O):
 -include $(AMBDEC_O:%.o=%.d)
@@ -47,7 +47,7 @@
 
 ambdec_cli:	LDLIBS += -lclthreads -ljack -lrt
 ambdec_cli:	$(AMBDEC_CLI_O)
-	g++ $(LDFLAGS) -o $@ $(AMBDEC_CLI_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(AMBDEC_CLI_O) $(LDLIBS)
 
 $(AMBDEC_CLI_O):
 -include $(AMBDEC_CLI_O:%.o=%.d)
@@ -55,10 +55,10 @@
 
 
 install:	all
-	/usr/bin/install -m 755 ambdec $(PREFIX)/bin
-	/usr/bin/install -m 755 ambdec_cli $(PREFIX)/bin
-	/usr/bin/install -d $(SHARED)/ambdec
-	/usr/bin/install -m 644 ../share/* $(SHARED)/ambdec
+	/usr/bin/install -D -m 755 ambdec $(DESTDIR)$(PREFIX)/bin/ambdec
+	/usr/bin/install -m 755 ambdec_cli $(DESTDIR)$(PREFIX)/bin/ambdec_cli
+	/usr/bin/install -d $(DESTDIR)$(SHARED)/ambdec
+	/usr/bin/install -m 644 ../share/* $(DESTDIR)$(SHARED)/ambdec
 
 
 clean:
