--- Makefile.orig	2007-05-29 14:45:07.000000000 +0200
+++ Makefile	2007-05-29 14:54:41.000000000 +0200
@@ -1,8 +1,8 @@
 
 # Location of Vamp SDK
 #
-VAMPDIR		= ../vamp-plugin-sdk
-VAMPLIBDIR	= $(VAMPDIR)/vamp-sdk
+VAMPDIR		= `pkg-config --cflags vamp-sdk`
+VAMPLIBDIR	= `pkg-config --libs vamp-sdk`
 
 # Location of our plugins
 #
@@ -10,7 +10,7 @@
 
 # Compile flags
 #
-CXXFLAGS	:= $(CXXFLAGS) -DNDEBUG -O2 -march=pentium3 -mfpmath=sse -ffast-math -Wall -I$(VAMPDIR) -I.
+CXXFLAGS	:= $(CXXFLAGS) -DNDEBUG -ffast-math -fPIC -DPIC -Wall -I$(VAMPDIR) -I.
 
 # Libraries required for the plugins.  Note that we can (and actively
 # want to) statically link libstdc++, because our plugin exposes only
@@ -22,7 +22,7 @@
 
 # Flags required to tell the compiler to make a dynamically loadable object
 #
-PLUGIN_LDFLAGS	= -shared -Wl,-Bsymbolic -static-libgcc
+PLUGIN_LDFLAGS	= -shared -Wl,-Bsymbolic -static-libgcc -fPIC
 
 # File extension for a dynamically loadable object
 #
