--- Makefile.orig	2006-05-16 15:13:49.340714007 +0200
+++ Makefile	2006-05-16 15:14:52.426123567 +0200
@@ -46,11 +46,11 @@
 
 
 install:	$(CLALSADRV_MIN)
-	/usr/bin/install -m 644 $(CLALSADRV_H) $(PREFIX)/include
-	/usr/bin/install -m 755 $(CLALSADRV_MIN) $(PREFIX)/$(LIBDIR)
-	/sbin/ldconfig -n $(PREFIX)/$(LIBDIR)
-	ln -sf $(CLALSADRV_MIN) $(PREFIX)/$(LIBDIR)/$(CLALSADRV_SO)
-
+	install -d $(DESTDIR)$(PREFIX)/include $(DESTDIR)$(PREFIX)/lib
+	install -m 644 $(CLALSADRV_H) $(DESTDIR)$(PREFIX)/include
+	install -m 755 $(CLALSADRV_MIN) $(DESTDIR)$(PREFIX)/lib
+	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
+	ln -sf $(CLALSADRV_MIN) $(DESTDIR)$(PREFIX)/lib/$(CLALSADRV_SO)
 
 clean:
 	/bin/rm -f *~ *.o *.a *.d *.so.* loopback
