--- src/SConscript.orig	2007-07-21 13:59:44.000000000 +0200
+++ src/SConscript	2007-07-21 14:02:01.000000000 +0200
@@ -325,7 +325,7 @@
 	env.Append(LIBS = 'shell32')
 
 env.Append(CXXFLAGS = ' -DQT3_SUPPORT -DQT3_SUPPORT_WARNINGS -DQT_THREAD_SUPPORT -DQT_SHARED -DQT_TABLET_SUPPORT') #Stolen from Mixxx's build output
-env.Append(CXXFLAGS = ' -I$QTDIR/include/Qt3Support -I$QTDIR/include/QtCore -I$QTDIR/include/QtGui -I$QTDIR/include/QtXml -I$QTDIR/include/QtOpenGl -I$QTDIR/include/Qt -I"$VCINSTALLDIR/include/atl"')
+env.Append(CXXFLAGS = ' -I/usr/include/qt4 -I/usr/include/qt4/Qt3Support -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtXml -I/usr/include/qt4/QtOpenGl -I/usr/include/qt4/Qt -I"$VCINSTALLDIR/include/atl"')
 
 if not platform == 'win32':
 	env.Append(CXXFLAGS = ' -pipe -O3 -pipe -Wall -W -g -D_REENTRANT') # omghax
--- src/qt4.py.orig	2007-07-21 15:29:25.000000000 +0200
+++ src/qt4.py	2007-07-21 15:31:06.000000000 +0200
@@ -246,7 +246,7 @@
 		QTDIR  = _detect(env),
 		QT4_BINPATH = os.path.join('$QTDIR', 'bin'),
 		QT4_CPPPATH = os.path.join('$QTDIR', 'include'),
-		QT4_LIBPATH = os.path.join('$QTDIR', 'lib'),
+		QT4_LIBPATH = os.path.join('/usr/lib/qt4'),
 		# TODO: This is not reliable to QTDIR value changes but needed in order to support '-qt4' variants
 		QT4_MOC = locateQt4Command(env,'moc', env['QTDIR']),
 		QT4_UIC = locateQt4Command(env,'uic', env['QTDIR']),
