--- mixxx-9999/build/depends.py
+++ mixxx-9999/build/depends.py
@@ -862,10 +862,8 @@ class MixxxCore(Feature):
         if build.toolchain_is_gnu:
             # Default GNU Options
             # TODO(XXX) always generate debugging info?
-            build.env.Append(CCFLAGS='-pipe')
             build.env.Append(CCFLAGS='-Wall')
             build.env.Append(CCFLAGS='-Wextra')
-            build.env.Append(CCFLAGS='-g')
 
             # Check that g++ is present (yeah, SCONS is a bit dumb here)
             # returns a non zeros return code if g++ is found
