=== modified file 'mixxx/plugins/soundsourcem4a/m4a/mp4-mixxx.cpp'
--- mixxx/plugins/soundsourcem4a/m4a/mp4-mixxx.cpp	2010-12-31 21:46:37 +0000
+++ mixxx/plugins/soundsourcem4a/m4a/mp4-mixxx.cpp	2012-04-08 18:12:25 +0000
@@ -163,8 +163,12 @@
     //         << neaac_cfg->dontUpSampleImplicitSBR;
     faacDecSetConfiguration(priv->decoder, neaac_cfg);
 
-    /* open mpeg-4 file */
+    /* open mpeg-4 file, check for >= ver 1.9.1 */
+    #if MP4V2_PROJECT_version_hex <= 0x00010901
     priv->mp4.handle = MP4Read(ip_data->filename, 0);
+    #else
+    priv->mp4.handle = MP4Read(ip_data->filename);
+    #endif
     if (!priv->mp4.handle) {
         qDebug() << "MP4Read failed";
         goto out;

