=== modified file 'mixxx/src/SConscript'
--- mixxx/src/SConscript	2012-01-23 11:25:21 +0000
+++ mixxx/src/SConscript	2012-05-05 00:05:56 +0000
@@ -219,7 +219,6 @@
 		env.Alias('install', keyboardmappings)
 		if int(flags['ladspa']):
 			env.Alias('install', ladspapresets)
-		env.Alias('install', docs)
 		env.Alias('install', dotdesktop)
 		env.Alias('install', icon)
 		env.Alias('install', promotracks)

