diff -r b6a04c79b54e -r 35a25c975b9f src/ghostess.c
--- a/src/ghostess.c	Fri Apr 27 17:15:09 2007 +0400
+++ b/src/ghostess.c	Fri Apr 27 17:17:09 2007 +0400
@@ -232,7 +232,7 @@ audio_callback(jack_nframes_t nframes, v
     void* midi_port_buf = jack_port_get_buffer(midi_input_port, nframes);
     jack_midi_event_t jack_midi_event;
     jack_nframes_t jack_midi_event_index = 0;
-    jack_nframes_t jack_midi_event_count = jack_midi_get_event_count(midi_port_buf, nframes);
+    jack_nframes_t jack_midi_event_count = jack_midi_get_event_count(midi_port_buf);
     static snd_seq_event_t jack_seq_event_holder[3];
     snd_seq_event_t *jack_seq_event = NULL, *osc_seq_event = NULL;
     int had_midi_overflow = 0;
@@ -258,7 +258,7 @@ audio_callback(jack_nframes_t nframes, v
 
             int count;
 
-            jack_midi_event_get(&jack_midi_event, midi_port_buf, jack_midi_event_index, nframes);
+            jack_midi_event_get(&jack_midi_event, midi_port_buf, jack_midi_event_index);
             jack_midi_event_index++;
 
             jack_seq_event = jack_seq_event_holder;
