--- Makefile.orig	2011-07-29 17:56:20.803333050 +0300
+++ Makefile	2011-07-29 17:56:53.698335388 +0300
@@ -15,6 +15,9 @@
 	@echo "build finish, now run make install"
 
 install:
+	install -d $(DESTDIR)$(BIN_DIR)
+	install -d $(DESTDIR)$(DESKAPPS_DIR)
+	install -d $(DESTDIR)$(PIXMAPS_DIR)
 	install gxtuner $(DESTDIR)$(BIN_DIR)
 	install gxtuner.desktop $(DESTDIR)$(DESKAPPS_DIR)
 	install gxtuner.png $(DESTDIR)$(PIXMAPS_DIR)
