--- Makefile.orig	2006-05-16 17:55:31.207806050 +0200
+++ Makefile	2006-05-16 17:59:22.408658179 +0200
@@ -64,10 +64,12 @@
 
 
 install:	aeolus aeolus_x11.so aeolus_txt.so 
-	/usr/bin/install -m 755 aeolus $(PREFIX)/bin
-	/usr/bin/install -m 755 aeolus_x11.so $(PREFIX)/$(LIBDIR)
-	/usr/bin/install -m 755 aeolus_txt.so $(PREFIX)/$(LIBDIR)
-	/sbin/ldconfig $(PREFIX)/$(LIBDIR)
+	install -d $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/lib
+	install -m 755 aeolus $(DESTDIR)$(PREFIX)/bin
+	install -m 755 aeolus_x11.so $(DESTDIR)$(PREFIX)/lib
+	install -m 755 aeolus_txt.so $(DESTDIR)$(PREFIX)/lib
+	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
+
 
 clean:
 	/bin/rm -f *~ *.o *.d *.a *.so aeolus
