--- Makefile.org	2006-11-30 22:34:59.000000000 +0100
+++ Makefile	2006-11-30 22:36:01.000000000 +0100
@@ -65,10 +65,11 @@
 
 install:	aeolus aeolus_x11.so aeolus_txt.so 
 	/usr/bin/install -d $(PREFIX)/$(LIBDIR)
-	/usr/bin/install -m 755 aeolus $(PREFIX)/bin
-	/usr/bin/install -m 755 aeolus_x11.so $(PREFIX)/$(LIBDIR)
-	/usr/bin/install -m 755 aeolus_txt.so $(PREFIX)/$(LIBDIR)
-	/sbin/ldconfig $(PREFIX)/$(LIBDIR)
+	install -d $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/lib
+	install -m 755 aeolus $(DESTDIR)$(PREFIX)/bin
+	install -m 755 aeolus_x11.so $(DESTDIR)$(PREFIX)/lib
+	install -m 755 aeolus_txt.so $(DESTDIR)$(PREFIX)/lib
+	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
 
 clean:
 	/bin/rm -f *~ *.o *.d *.a *.so aeolus
