diff -Naur freecycle/src/soundplayer.cpp freecycle.neu/src/soundplayer.cpp
--- freecycle/src/soundplayer.cpp	2006-03-20 18:15:22.000000000 +0100
+++ freecycle.neu/src/soundplayer.cpp	2006-03-24 12:34:17.577192250 +0100
@@ -74,7 +74,7 @@
         bufferLR_F=new float[buffer_len*2];
         bufferPointerL=0;
         bufferPointerR=0;
-        bufferPointersLR=new (float*)[2];
+        bufferPointersLR=new float*[2];  /* () removed due to gcc >= 3.4 compatibility */
         #ifdef HAS_OSS
             devdsp_audio_device=0;
         #endif
