--- Makefile.orig	2006-10-08 16:59:35.000000000 +0200
+++ Makefile	2006-11-22 15:59:20.000000000 +0100
@@ -48,10 +48,11 @@
 
 install:	$(CLXCLIENT_MIN)
 	/usr/bin/install -d $(PREFIX)/$(LIBDIR)
-	/usr/bin/install -m 644 $(CLXCLIENT_H) $(PREFIX)/include
-	/usr/bin/install -m 755 $(CLXCLIENT_MIN) $(PREFIX)/$(LIBDIR)
-	/sbin/ldconfig -n $(PREFIX)/$(LIBDIR)
-	ln -sf $(CLXCLIENT_MIN) $(PREFIX)/$(LIBDIR)/$(CLXCLIENT_SO)
+	install -d $(DESTDIR)$(PREFIX)/include $(DESTDIR)$(PREFIX)/lib
+	install -m 644 $(CLXCLIENT_H) $(DESTDIR)$(PREFIX)/include
+	install -m 755 $(CLXCLIENT_MIN) $(DESTDIR)$(PREFIX)/lib
+	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
+	ln -sf $(CLXCLIENT_MIN) $(DESTDIR)$(PREFIX)/lib/$(CLXCLIENT_SO)
 
 
 clean:
