--- Makefile.orig	2012-10-25 10:31:19.847614150 +0100
+++ Makefile	2012-10-25 19:31:05.897383425 +0100
@@ -17,7 +17,6 @@
 # ADD_DEF=
 
 CC=gcc 
-LD=gcc
 INSTALL=install
 PREFIX=/usr/local
 BINDIR=$(PREFIX)/bin
@@ -40,7 +39,7 @@
 GCF = -DETCDIR=\\\"$(ETCDIR)\\\" `pkg-config --cflags gtk+-2.0`
 
 all:	
-	$(MAKE) target CFLAGS="-O2 -ffast-math -fomit-frame-pointer $(GCF) $(ADD_DEF)"
+	$(MAKE) target CFLAGS="$(CFLAGS) -ffast-math $(GCF) $(ADD_DEF)"
 
 debug:
 	$(MAKE) target CFLAGS="-g -Wall -W -Wno-unused-parameter -D__NO_MATH_INLINES $(GCF) $(ADD_DEF)"
@@ -76,7 +75,7 @@
 
 target:  $(OBJ) postfish-wisdomrc
 	./touch-version
-	$(LD) $(OBJ) $(CFLAGS) -o postfish $(LIBS) `pkg-config --libs gtk+-2.0` -lpthread -lfftw3f -lm
+	$(CC) $(LDFLAGS) $(OBJ) $(CFLAGS) -o postfish $(LIBS) `pkg-config --libs gtk+-2.0` -lpthread -lfftw3f -lm
 
 install: target
 	$(INSTALL) -d -m 0755 $(BINDIR)
