--- ac3jack-2.0.3/src/jack_audio_driver.cpp	2009-12-22 12:22:15.449869661 +0100
+++ ac3jack-2.0.3/src/jack_audio_driver.cpp	2009-12-22 12:22:34.356707546 +0100
@@ -25,6 +25,7 @@
 
 #include "jack_audio_driver.hpp"
 #include "engine.hpp"
+#include <stdio.h>
 
 using namespace AC3JACK;
 using namespace PBD;

--- ac3jack-2.0.3/src/control_osc.cpp	2009-12-22 13:03:37.723470941 +0100
+++ ac3jack-2.0.3/src/control_osc.cpp	2009-12-22 13:03:02.986366348 +0100
@@ -34,6 +34,7 @@
 
 #include <lo/lo.h>
 #include <sigc++/sigc++.h>
+#include <string.h>
 using namespace SigC;
 
 using namespace AC3JACK;
