--- fftscope.c.orig	2006-09-30 01:30:44.000000000 +0200
+++ fftscope.c	2006-09-30 01:12:43.000000000 +0200
@@ -249,6 +249,12 @@
 	return 1;
 }
 
+/* Are we running? */
+static int fftscope_running()
+{
+	return running;
+}
+
 
 /* Shutdown function. Should deallocate stuff here too */
 static void shutdown_fftscope()
@@ -258,12 +264,6 @@
 }
 
 
-/* Are we running? */
-static int fftscope_running()
-{
-	return running;
-}
-
 
 /* The C struct that contains all our functions */
 static scope_plugin fftscope_plugin;
