--- install-init.sh-orig	2007-10-03 13:24:06.000000000 +0200
+++ install-init.sh	2009-07-23 00:50:51.000000000 +0200
@@ -23,13 +23,12 @@
 #
 
 if test ! -f ./building-debian ; then
-  mkdir -p /etc/init.d
-  cp rivendell /etc/init.d/rivendell
-  mkdir -p /etc/sysconfig
-  cp rivendell.sys /etc/sysconfig/rivendell
-  mkdir -p /var/run/rivendell
-  chmod 777 /var/run/rivendell
-  ldconfig
+  mkdir -p ${DESTDIR}/etc/init.d
+  cp rivendell ${DESTDIR}/etc/init.d/rivendell
+  mkdir -p ${DESTDIR}/etc/sysconfig
+  cp rivendell.sys ${DESTDIR}/etc/sysconfig/rivendell
+  mkdir -p ${DESTDIR}/var/run/rivendell
+  chmod 777 ${DESTDIR}/var/run/rivendell
 fi
 
 chmod 4755 $1/bin/caed
