Index: euses.c
===================================================================
--- euses.c	(revision 12)
+++ euses.c	(working copy)
@@ -130,7 +130,6 @@
 /* param: *portdir where to put PORTDIR */
 static char *getportdir(char *portdir)
 {
-	FILE *fd;
 	struct stat *finfo;
 	char *path;
 
@@ -142,17 +141,7 @@
 	} else {
 
 	/* or load it from make.{globals,conf} */
-		/* Open a pipe. */
-		fd = popen(" . " SYSCONFDIR "/make.globals;" " . "
-			SYSCONFDIR "/make.conf" ";"
-			"echo -n ${PORTDIR}", "r");
-
-		/* Get the first line from the pipe */
-		if (fgets(portdir, (int)FILENAME_MAX, fd) == NULL)
-			doerror(FILE_ERR, "PORTDIR", NULL);
-
-		/* Close the pipe. */
-		pclose(fd);
+		pipe_gentoo_makeconf_var("PORTDIR",portdir);
 	}
 	/* Turn the portdir string into a path */
 	stat(realpath(portdir, path), finfo);
@@ -166,6 +155,27 @@
 	return portdir;
 }
 
+/* get the content of a given VARIABLE from make.{globals,conf} 
+ * param: char *gentoo_make_var -- VARIABLE from make.{globals,conf}
+ * param: char *dir -- where to put the content of the found VARIALE */
+void pipe_gentoo_makeconf_var(char *gentoo_make_var, char *dir)
+{
+	FILE *fd;
+	/* Open a pipe. */
+	char *str = (char *) malloc(FILENAME_MAX);
+	snprintf(str, FILENAME_MAX," . %s/make.globals; . %s/make.conf;echo -n $%s",
+		SYSCONFDIR, SYSCONFDIR, gentoo_make_var);
+	fd = popen(str, "r");
+	free(str);
+	
+	/* Get the first line from the pipe */
+	if (fgets(dir, (int)FILENAME_MAX, fd) == NULL)
+		doerror(FILE_ERR, gentoo_make_var, NULL);
+
+	/* Close the pipe. */
+	pclose(fd);
+}
+
 /* fill the glob_t structure with matching files
  * param: char *profiles_glob  -- .../profiles/[foo].desc
  * param: char *profiles_desc_glob -- .../profiles/desc/[foo].desc
@@ -228,24 +238,13 @@
 * return: **overlays where to put PORTDIR_OVERLAY */
 static char **get_overlay_dir(short *overlay_entries)
 {
-	FILE *fd = NULL;
 	char *buffer = malloc(FILENAME_MAX);
 	/* tokenize overlays (PORTDIR_OVERLAY) */	
 	char **overlays;
  
-	/* load it from make.{globals,conf} */
-	/* Open a pipe. */
-	fd = popen(" . " SYSCONFDIR "/make.globals;" " . "
-		SYSCONFDIR "/make.conf" ";"
-		"echo -n ${PORTDIR_OVERLAY}", "r");
-	/* Get the first line from the pipe */
-	if (fgets(buffer, (int)FILENAME_MAX, fd) == NULL)
-		doerror(FILE_ERR, "PORTDIR_OVERLAY", NULL);
-
-	/* Close the pipe. */
-	pclose(fd);
+	/* load overlays from make.{globals,conf} */
+	pipe_gentoo_makeconf_var("PORTDIR_OVERLAY",buffer);
 	
-	
 	return overlays = tokenizer(buffer,overlay_entries," ");
 }
 
Index: euses.h
===================================================================
--- euses.h	(revision 12)
+++ euses.h	(working copy)
@@ -107,6 +107,8 @@
 
 void fill_profiles_glob(int *desc_files_to_search, char *dir, char *profiles_glob, char *profiles_desc_glob);
 
+void pipe_gentoo_makeconf_var(char *gentoo_make_var, char *dir);
+
 /* TODO: Struct typedefs: */
 
 typedef struct {
