*** Makefile.orig	2008-01-07 00:18:20.000000000 +0100
--- Makefile	2008-01-07 23:23:54.000000000 +0100
***************
*** 15,21 ****
  #  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  
  
! PREFIX = /usr/local
  SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
  LIBDIR = lib$(SUFFIX)
  VERSION = 0.7.0
--- 15,21 ----
  #  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  
  
! PREFIX = /usr
  SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
  LIBDIR = lib$(SUFFIX)
  VERSION = 0.7.0
***************
*** 26,32 ****
  
  # Uncomment if you use a libsndfile version without support for setting/getting the Ambisonic flags.
  #
! #CPPFLAGS += -DNOAMBIS=1
  
  
  
--- 26,32 ----
  
  # Uncomment if you use a libsndfile version without support for setting/getting the Ambisonic flags.
  #
! CPPFLAGS += -DNOAMBIS=1
  
  
  
***************
*** 79,90 ****
  
  
  install:	all
! 	/usr/bin/install -d $(PREFIX)/$(LIBDIR)
! 	/usr/bin/install -m 755 aeolus-ym $(PREFIX)/bin
! 	/usr/bin/install -m 755 mkwavex $(PREFIX)/bin
! 	/usr/bin/install -m 755 aeolus-ym_x11.so $(PREFIX)/$(LIBDIR)
! 	/usr/bin/install -m 755 aeolus-ym_txt.so $(PREFIX)/$(LIBDIR)
! 	/sbin/ldconfig $(PREFIX)/$(LIBDIR)
  
  clean:
  	/bin/rm -f *~ *.o *.d *.a *.so aeolus-ym mkwavex
--- 79,90 ----
  
  
  install:	all
! 	install -d $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/lib
! 	install -m 755 aeolus-ym $(DESTDIR)$(PREFIX)/bin
! 	install -m 755 mkwavex $(DESTDIR)$(PREFIX)/bin
! 	install -m 755 aeolus-ym_x11.so $(DESTDIR)$(PREFIX)/lib
! 	install -m 755 aeolus-ym_txt.so $(DESTDIR)$(PREFIX)/lib
! 	/sbin/ldconfig -n $(DESTDIR)$(PREFIX)/lib
  
  clean:
  	/bin/rm -f *~ *.o *.d *.a *.so aeolus-ym mkwavex
