Index: phat2/configure.ac
===================================================================
--- phat2/configure.ac	(revision 117)
+++ phat2/configure.ac	(working copy)
@@ -20,10 +20,12 @@
     [if test x$enable_debug = xyes; then with_debug=yes ; fi])
 
 if test x$with_debug = xno; then
-    CFLAGS="-O3"
+    if test -z $CFLAGS; then 
+        CFLAGS="-O3"
+    fi
     AC_DEFINE(DEBUG, 0, [[whether to display debugging output or not]])
 else
-    CFLAGS="-g"
+    CFLAGS="$CFLAGS -g"
     AC_DEFINE(DEBUG, 1, [[whether to display debugging output or not]])
 fi
 
