--- configure.orig	2007-06-29 22:00:47.000000000 +0300
+++ configure	2007-06-29 22:05:35.000000000 +0300
@@ -40,10 +40,11 @@
 			/usr/local/Trolltech/Qt-4.1.2 \
 			/usr/local/Trolltech/Qt-4.1.1 \
 			/usr/local/Trolltech/Qt-4.1.0 \
-			/usr \
+			/usr/lib \
+			/usr/lib/qt4 \
 			; do
 		if test "x$ac_qtlib" = "x"; then
-			if test -f $X/lib/libQtCore.a -o -f $X/lib/libQtCore.so -o -d $X/lib/QtCore.framework -o -f $X/lib/libQtCore.dylib; then
+			if test -f $X/libQtCore.a -o -f $X/libQtCore.so -o -d $X/QtCore.framework -o -f $X/libQtCore.dylib; then
 				ac_qtlib=$X
 				QTDIR=$X
 				echo "   Found QT4 libraries in $X"
@@ -55,7 +56,7 @@
 	echo ""
 	ac_qtlib=""
 	if test "x$ac_qtlib" = "x"; then
-		if test -f $QTDIR/lib/libQtCore.so -o -f $QTDIR/lib/libQtCore.a -o -f $QTDIR/lib/libQtCore_debug.so -o -f $QTDIR/lib/libQtCore_debug.a -o -d $QTDIR/lib/QtCore.framework -o -f $QTDIR/lib/libQtCore.dylib; then
+		if test -f $QTDIR/libQtCore.so -o -f $QTDIR/libQtCore.a -o -f $QTDIR/libQtCore_debug.so -o -f $QTDIR/libQtCore_debug.a -o -d $QTDIR/QtCore.framework -o -f $QTDIR/libQtCore.dylib; then
 			ac_qtlib=$QTDIR
 			echo " * Searching for QT library in $QTDIR [ OK ]"
 		else
@@ -71,11 +72,11 @@
 fi
 
 
-QMAKE_BIN=$QTDIR/bin/qmake
+QMAKE_BIN=/usr/bin/qmake
 
 if [ -f $QTDIR/bin/qmake-qt4 ]
 then
-	QMAKE_BIN=$QTDIR/bin/qmake-qt4
+	QMAKE_BIN=/usr/bin/qmake-qt4
 fi
 
 echo ""
