--- muse/muse/audiotrack.cpp	2008-09-17 14:31:37.995325305 +0200
+++ muse/muse/audiotrack.cpp	2008-09-17 18:42:20.000000000 +0200
@@ -21,10 +21,10 @@
 // By T356. For caching jack in/out routing names BEFORE file save. 
 // Jack often shuts down during file save, causing the routes to be lost in the file.
 // cacheJackRouteNames() is ONLY called from MusE::save() in app.cpp
-typedef std::multimap <int, QString> jackRouteNameMap;
-std::map <unsigned int, jackRouteNameMap > jackRouteNameCache;
-typedef std::multimap <int, QString>::const_iterator ciJackRouteNameMap;
-typedef std::map <unsigned int, jackRouteNameMap>::const_iterator ciJackRouteNameCache;
+typedef std::multimap <unsigned long, QString> jackRouteNameMap;
+std::map <unsigned long, jackRouteNameMap > jackRouteNameCache;
+typedef std::multimap <unsigned long, QString>::const_iterator ciJackRouteNameMap;
+typedef std::map <unsigned long, jackRouteNameMap>::const_iterator ciJackRouteNameCache;
 void cacheJackRouteNames()
 {
     jackRouteNameCache.clear();
@@ -37,7 +37,7 @@
         jackRouteNameMap rm = jackRouteNameMap();
         for(ciRoute r = rl->begin(); r != rl->end(); ++r) 
           rm.insert(std::pair<const int, QString>(r->channel, r->name()));
-        jackRouteNameCache.insert(std::pair<const unsigned int, jackRouteNameMap>((unsigned int)(*iai), rm));
+        jackRouteNameCache.insert(std::pair<const unsigned long, jackRouteNameMap>((unsigned long)(*iai), rm));
       }                            
     }
     const OutputList* ol = song->outputs();
@@ -49,7 +49,7 @@
         jackRouteNameMap rm = jackRouteNameMap();
         for(ciRoute r = rl->begin(); r != rl->end(); ++r) 
           rm.insert(std::pair<const int, QString>(r->channel, r->name()));
-        jackRouteNameCache.insert(std::pair<const unsigned int, jackRouteNameMap>((unsigned int)(*iao), rm));
+        jackRouteNameCache.insert(std::pair<const unsigned long, jackRouteNameMap>((unsigned long)(*iao), rm));
       }                            
     }
 }
@@ -1082,7 +1082,7 @@
       {
       QString n;
       if (type() == Track::AUDIO_INPUT) {
-                ciJackRouteNameCache circ = jackRouteNameCache.find((unsigned int)this);
+                ciJackRouteNameCache circ = jackRouteNameCache.find((unsigned long)this);
                 if(circ != jackRouteNameCache.end())
                 {
                   jackRouteNameMap rm = circ->second;
@@ -1102,7 +1102,7 @@
             }
       if(type() == Track::AUDIO_OUTPUT) 
       {
-        ciJackRouteNameCache circ = jackRouteNameCache.find((unsigned int)this);
+        ciJackRouteNameCache circ = jackRouteNameCache.find((unsigned long)this);
         if(circ != jackRouteNameCache.end())
         {
           jackRouteNameMap rm = circ->second;
