--- jack-keyboard.c.orig	2007-12-15 23:28:50.000000000 +0100
+++ jack-keyboard.c	2007-12-15 23:30:59.000000000 +0100
@@ -202,10 +202,10 @@
 
 	port_buffer = jack_port_get_buffer(input_port, nframes);
 
-	events = jack_midi_get_event_count(port_buffer, nframes);
+	events = jack_midi_get_event_count(port_buffer);
 
 	for (i = 0; i < events; i++) {
-		read = jack_midi_event_get(&event, port_buffer, i, nframes);
+		read = jack_midi_event_get(&event, port_buffer, i);
 		if (read) {
 			fprintf(stderr, "jack_midi_event_get failed, RECEIVED NOTE LOST.\n");
 			continue;
@@ -222,7 +222,7 @@
 
 	port_buffer = jack_port_get_buffer(output_port, nframes);
 
-	jack_midi_clear_buffer(port_buffer, nframes);
+	jack_midi_clear_buffer(port_buffer);
 
 	while (jack_ringbuffer_read_space(ringbuffer)) {
 		read = jack_ringbuffer_read(ringbuffer, (char *)&ev, sizeof(ev));
@@ -232,7 +232,7 @@
 			continue;
 		}
 
-		buffer = jack_midi_event_reserve(port_buffer, 0, ev.len, nframes);
+		buffer = jack_midi_event_reserve(port_buffer, 0, ev.len);
 
 		if (buffer == NULL) {
 			fprintf(stderr, "jack_midi_event_reserve failed, NOTE LOST.\n");
