--- jack_capture-0.9.44/Makefile.old	2010-05-01 23:31:56.000000000 +0100
+++ jack_capture-0.9.44/Makefile	2010-05-02 00:56:53.000000000 +0100
@@ -1,14 +1,14 @@
 
-DESTDIR =
-prefix = /usr/local
+DESTDIR ?=
+prefix ?= /usr/local
 bindir = $(prefix)/bin
 
 VERSION=0.9.44
 
-CC=gcc
-CPP=g++
+CC ?= gcc
+CPP ?= g++
 
-OPTIMIZE=-O3 -march=native
+OPTIMIZE ?=-O3 -march=native
 #OPTIMIZE=-O0 -g
 
 # __SMP__ is defined by default in msAtomic.h
