diff -urN dssi_convolve-0.0.4/Makefile dssi_convolve-0.0.4.new/Makefile
--- dssi_convolve-0.0.4/Makefile	2005-11-28 03:49:57.000000000 +0100
+++ dssi_convolve-0.0.4.new/Makefile	2006-03-10 22:25:27.100174628 +0100
@@ -5,7 +5,7 @@
 
 .PHONY: all
 all:
-	for n in 1 2 4 6; do echo; CHANNELS=$$n make -f Makefile.plugin || exit 42; done
+	for n in 1 2 4 6; do echo;  PREFIX=$(PREFIX) CHANNELS=$$n make -f Makefile.plugin || exit 42; done
 	echo; make PREFIX=$(PREFIX) -f Makefile.ui
 
 .PHONY: install
diff -urN dssi_convolve-0.0.4/Makefile.plugin dssi_convolve-0.0.4.new/Makefile.plugin
--- dssi_convolve-0.0.4/Makefile.plugin	2005-11-28 14:02:55.000000000 +0100
+++ dssi_convolve-0.0.4.new/Makefile.plugin	2006-03-10 22:23:02.632074031 +0100
@@ -1,6 +1,7 @@
-PLUGIN_COMPILE_FLAGS = -Wall `pkg-config dssi --cflags` `pkg-config sndfile --cflags` `pkg-config samplerate --cflags` -I/usr/local/include -D_GNU_SOURCE -O2
+INCLUDE=$(PREFIX)/include
+CXXFLAGS = $(CFLAGS) -Wall `pkg-config dssi --cflags` `pkg-config sndfile --cflags` `pkg-config samplerate --cflags` -I$(INCLUDE) -D_GNU_SOURCE 
 
-PLUGIN_COMPILE_FLAGS += -DCHANNELS=$(CHANNELS) -DLADSPA_ID=290$(CHANNELS) -DNAME='"Conv$(CHANNELS)ch."'
+CXXFLAGS += -DCHANNELS=$(CHANNELS) -DLADSPA_ID=290$(CHANNELS) -DNAME='"Conv$(CHANNELS)ch."'
 
 PLUGIN_LINK_FLAGS    = -nostartfiles -shared `pkg-config sndfile --libs` `pkg-config samplerate --libs` -lconvolve -lstdc++ -lm -pthread
 
@@ -17,7 +18,7 @@
 
 #.PHONY: $(PLUGIN_OBJECTS)
 $(PLUGIN_OBJECT):  $(PLUGIN_SOURCE)
-	$(CC) -c -o $(PLUGIN_OBJECT) $(PLUGIN_COMPILE_FLAGS) $(PLUGIN_SOURCE)
+	$(CC) -c -o $(PLUGIN_OBJECT) $(CXXFLAGS) $(PLUGIN_SOURCE)
  
 
 .PHONY: clean
@@ -26,7 +27,7 @@
 
 .PHONY: install
 install: $(PLUGIN_TARGET)
-	install -d $(PREFIX)/lib/dssi/
-	install -d $(PREFIX)/share/dssi_convolve/
-	install $(PLUGIN_TARGET) $(PREFIX)/lib/dssi/
+	install -d $(DESTDIR)$(PREFIX)/lib/dssi/
+	install -d $(DESTDIR)$(PREFIX)/share/dssi_convolve/
+	install $(PLUGIN_TARGET) $(DESTDIR)$(PREFIX)/lib/dssi/
 
diff -urN dssi_convolve-0.0.4/Makefile.ui dssi_convolve-0.0.4.new/Makefile.ui
--- dssi_convolve-0.0.4/Makefile.ui	2005-11-28 04:14:32.000000000 +0100
+++ dssi_convolve-0.0.4.new/Makefile.ui	2006-03-10 22:22:42.138605561 +0100
@@ -1,4 +1,4 @@
-GUI_COMPILE_FLAGS = -Wall `pkg-config dssi --cflags` `pkg-config gtk+-2.0 --cflags` `pkg-config libglade-2.0 --cflags` `pkg-config liblo --cflags` -D_GNU_SOURCE -g -DPREFIX=\"$(PREFIX)\"
+CXXFLAGS =  $(CFLAGS) -Wall `pkg-config dssi --cflags` `pkg-config gtk+-2.0 --cflags` `pkg-config libglade-2.0 --cflags` `pkg-config liblo --cflags` -D_GNU_SOURCE -g -DPREFIX=\"$(PREFIX)\"
 
 GUI_LINK_FLAGS    = `pkg-config libglade-2.0 --cflags --libs`  `pkg-config liblo --libs` `pkg-config gtk+ --libs` -export-dynamic
 
@@ -23,7 +23,7 @@
 	$(CC) -o $(GUI_TARGET) $(GUI_OBJECTS) $(GUI_LINK_FLAGS)
 
 $(GUI_OBJECTS): %.o: %.c
-	$(CC) -c $< $(GUI_COMPILE_FLAGS)
+	$(CC) -c $< $(CXXFLAGS)
 
 
 .PHONY: clean
@@ -32,15 +32,16 @@
 
 .PHONY: install
 install: $(GUI_TARGET)
-	install -d $(PREFIX)/lib/dssi/dssi_convolve/
-	install -d $(PREFIX)/share/dssi_convolve/
-	install $(GUI_TARGET) $(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET)
-	install $(GUI_GLADE) $(PREFIX)/share/dssi_convolve/
-	install $(GUI_LOGO) $(PREFIX)/share/dssi_convolve/
+	echo "hier das prefix $(PREFIX) "
+	install -d $(DESTDIR)$(PREFIX)/lib/dssi/dssi_convolve/
+	install -d $(DESTDIR)$(PREFIX)/share/dssi_convolve/
+	install $(GUI_TARGET)  $(DESTDIR)$(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET)
+	install $(GUI_GLADE)  $(DESTDIR)$(PREFIX)/share/dssi_convolve/
+	install $(GUI_LOGO)  $(DESTDIR)$(PREFIX)/share/dssi_convolve/
 .PHONY: install_links
 install_links:
-	$(foreach nch,$(CHANNELS), ln -s $(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET) $(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET_PREFIX)$(nch)ch$(GUI_TARGET_SUFFIX) || true;)
+	$(foreach nch,$(CHANNELS), ln -s $(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET) $(DESTDIR)$(PREFIX)/lib/dssi/dssi_convolve/$(GUI_TARGET_PREFIX)$(nch)ch$(GUI_TARGET_SUFFIX) || true;)
 
-	$(foreach nch,$(CHANNELS), ln -s $(PREFIX)/lib/dssi/dssi_convolve/ $(PREFIX)/lib/dssi/dssi_convolve$(nch)ch || true;)
+	$(foreach nch,$(CHANNELS), ln -s $(PREFIX)/lib/dssi/dssi_convolve/ $(DESTDIR)$(PREFIX)/lib/dssi/dssi_convolve$(nch)ch || true;)
 
 
