--- a/ardour-0.99/SConstruct	2005-09-24 04:53:13.000000000 +0200
+++ 7/ardour/SConstruct	2005-11-19 02:28:52.000000000 +0100
@@ -35,6 +35,7 @@
     BoolOption('NOARCH', 'Do not use architecture-specific compilation flags', 0),
     PathOption('PREFIX', 'Set the install "prefix"', '/usr/local'),
     BoolOption('VST', 'Compile with support for VST', 0),
+    PathOption('VSTPATH', 'set where xfst build can be found', ''),
     BoolOption('VERSIONED', 'Add version information to ardour/gtk executable name inside the build directory', 0),
     BoolOption('USE_SSE_EVERYWHERE', 'Ask the compiler to use x86/SSE instructions and also our hand-written x86/SSE optimizations when possible (off by default)', 0),
     BoolOption('BUILD_SSE_OPTIMIZATIONS', 'Use our hand-written x86/SSE optimizations when possible (off by default)', 0)
