--- tetraproc-0.6.2/source/Makefile.old	2010-04-12 17:11:36.000000000 +0100
+++ tetraproc-0.6.2/source/Makefile	2010-04-12 17:11:11.000000000 +0100
@@ -14,11 +14,11 @@
 #  along with this program; if not, write to the Free Software
 #  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-
-PREFIX = /usr/local
+CXX ?= g++
+PREFIX ?= /usr/local
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 SHARED = $(PREFIX)/share/tetraproc
-LIBDIR = lib$(SUFFIX)
+LIBDIR ?= lib$(SUFFIX)
 VERSION = 0.6.2
 DISTDIR = tetraproc-$(VERSION)
 CPPFLAGS += -O3 -Wall -MMD -MP -DVERSION=\"$(VERSION)\" -DSHARED=\"$(SHARED)\"
@@ -42,16 +42,16 @@
 tetraproc:	LDLIBS += -lsndfile -lfftw3f -lclxclient -lclthreads -ljack -lpng -lXft -lX11 -lrt 
 tetraproc:	LDFLAGS += -L/usr/X11R6/lib
 tetraproc:	$(TETRAPROC_O)
-	g++ $(LDFLAGS) -o $@ $(TETRAPROC_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(TETRAPROC_O) $(LDLIBS)
 
 $(TETRAPROC_O):
 -include $(TETRAPROC_O:%.o=%.d)
 
 
 install:	all
-	/usr/bin/install -m 755 tetraproc $(PREFIX)/bin
-	/usr/bin/install -d $(SHARED)
-	/usr/bin/install -m 644 ../share/* $(SHARED)
+	/usr/bin/install -D -m 755 tetraproc $(DESTDIR)$(PREFIX)/bin/tetraproc
+	/usr/bin/install -d $(DESTDIR)$(SHARED)
+	/usr/bin/install -m 644 ../share/* $(DESTDIR)$(SHARED)
 
 
 clean:
