--- buzztard-0.5.0/src/ui/edit/Makefile.am.old	2010-10-23 12:21:45.000000000 +0100
+++ buzztard-0.5.0/src/ui/edit/Makefile.am	2010-10-23 12:22:17.000000000 +0100
@@ -14,6 +14,11 @@
 
 noinst_LTLIBRARIES		= libbtedit.la
 noinst_HEADERS			= marshal.h
+libbtedit_la_LIBADD = \
+	$(top_builddir)/src/lib/core/libbuzztard-core.la \
+	$(top_builddir)/src/lib/ic/libbuzztard-ic.la \
+	$(BASE_DEPS_LIBS) $(GUI_DEPS_LIBS) $(BT_LIBS) $(LIBINTL) \
+	-lgstinterfaces-0.10
 libbtedit_la_SOURCES	= \
 	about-dialog.c about-dialog.h about-dialog-methods.h \
 	edit-application.c edit-application.h edit-application-methods.h \
