--- a/synti/zynaddsubfx/CMakeLists.txt.orig	2007-03-22 20:55:08.000000000 +0100
+++ b/synti/zynaddsubfx/CMakeLists.txt	2007-03-22 21:28:53.000000000 +0100
@@ -52,6 +52,16 @@
 #      Input/ALSAMidiIn.C
 #      Input/NULLMidiIn.C
 
+
+FIND_LIBRARY(FLTK NAMES fltk PATHS /usr/lib/fltk-1.1)
+set(FLTK ${FLTK})
+
+#FIND_PATH(FLTK_INCL_DIR  PATHS /usr/include/fltk-1.1/ )
+set(FLTK_INCL_DIR /usr/include/fltk-1.1/)
+
+FIND_PATH(CONFIG_INCL_DIR config.h PATHS ../../ )
+set(CONFIG_INCL_DIR ${CONFIG_INCL_DIR})
+
 add_library ( zynaddsubfx SHARED
       ${fltk_ui_FLTK_UI_SRCS}
       DSP/AnalogFilter.C
@@ -107,12 +117,12 @@
 
 set_target_properties ( zynaddsubfx
    PROPERTIES PREFIX ""
-   COMPILE_FLAGS "-O6 -fvisibility=hidden -DASM_F2I_YES -DOS_LINUX -DFFTW_VERSION_3"
+   COMPILE_FLAGS "-O6 -fvisibility=hidden -DASM_F2I_YES -DOS_LINUX -DFFTW_VERSION_3 -I${FLTK_INCL_DIR} -I${CONFIG_INCL_DIR}"
    )
 
 target_link_libraries( zynaddsubfx
       synti jack asound pthread fftw3 mxml
-      m z fltk
+      m z ${FLTK}
       )
 
 install_targets ( /${CMAKE_INSTALL_LIBDIR}/${MusE_INSTALL_NAME}/synthi/ zynaddsubfx )
