--- muse/app.cpp.orig	2006-03-28 08:35:41.780382572 +0200
+++ muse/app.cpp	2006-03-28 08:39:21.324653946 +0200
@@ -2715,6 +2715,7 @@
       switch (lash_event_get_type (event))
         {
         case LASH_Save_File:
+	  {
           /* save file */
           const char *name = lash_get_fqn (lash_event_get_string (event), "lash-project-muse.med");
           int ok = save (name, false);
@@ -2725,27 +2726,34 @@
                 museProject = QFileInfo(name).dirPath(true);
                 }
           lash_send_event (lash_client, event);
-          break;
+	  }
+	  break;
 
-        case LASH_Restore_File:
+        case LASH_Restore_File: 
+	  {
           /* load file */
           loadProjectFile (lash_get_fqn (lash_event_get_string (event), "lash-project-muse.med" ), false, true);
           lash_send_event (lash_client, event);
+	  }
           break;
 
         case LASH_Quit:
+	  {
           /* quit muse */
           std::cout << "Quit"
                     << std::endl;
           lash_event_destroy (event);
-          break;
+	  }
+	  break;
 
         default:
+	  {
           std::cout << "Recieved unknown LASH event of type "
                     << lash_event_get_type (event)
                     << std::endl;
           //lash_event_destroy (event);
-          break;
+	  }
+	  break;
         }
     }
 }
