--- configure.ac.old	2006-04-01 01:22:57.993331106 +0200
+++ configure.ac	2006-04-01 01:23:26.322419418 +0200
@@ -187,7 +187,7 @@
 ##############
 muse_enable_lash="no"
 AC_ARG_ENABLE(lash,
-[  --enable-lash         enable LASH support],[
+[  --enable-lash           enable LASH support],[
   case "$enableval" in
     "yes")
       muse_enable_lash="yes"
@@ -507,7 +507,7 @@
   MusE configured
 
   using rtcap:         $report_rtcap
-  LASH support:      $LASH_FOUND
+  LASH support:        $LASH_FOUND
   setuid root install: $SUIDINSTALL
   setuid root build:   $SUIDBUILD
   VST/win support:     $VST
