--- /etc/init.d/rtirq.orig	2006-04-22 18:39:11.000000000 +0200
+++ /etc/init.d/rtirq	2006-04-22 18:38:52.320802850 +0200
@@ -58,13 +58,17 @@
 function check_binaries() {
 	TOOLS=(chrt pidof)
         for tool in "${TOOLS[@]}" ;do
-                if ! type $tool &> /dev/null ; then
+                if ! type ${tool} &> /dev/null ; then
                     eerror "Install $tool and run this script again"
                     return 1
                 fi
-		[ "$tool" == "chrt" ] && RTIRQ_CHRT="`type -p $tool 2>/dev/null`"
-		[ "$tool" == "pidof" ] && RTIRQ_PIDOF="`type -p $tool 2>/dev/null`"
+		[ "${tool}" == "chrt" ] && RTIRQ_CHRT="$(type -p ${tool} 2>/dev/null)"
+		[ "${tool}" == "pidof" ] && RTIRQ_PIDOF="$(type -p ${tool} 2>/dev/null)"
         done
+	# a user reported troubles with above "getting paths" for
+	# pidof. So here is a workaround for it
+	[ "${#RTIRQ_CHRT}" == "0" ] && RTIRQ_CHRT="chrt"
+	[ "${#RTIRQ_PIDOF}" == "0" ] && RTIRQ_PIDOF="pidof"
 	return 0
 }
 
