--- init.d/rtirq.sh	2006-09-15 01:01:33.000000000 +0200
+++ init.d/rtirq.sh	2006-09-15 01:07:04.000000000 +0200
@@ -104,7 +104,12 @@
 	fi
 	# And now the proper threading prioritization...
 	PID=`${RTIRQ_PIDOF} "IRQ ${IRQ}"`
-	for lPID in ${PID};do
+	PID=($PID)
+	
+	loopcnt=0
+	for lPID in ${PID[*]};do
+		loopcnt=$(($loopcnt + 1))
+	
 	if [ $((${lPID})) -gt 0 -a -z "`echo ${RTIRQ_TRAIL} | grep :${IRQ}:`" ]
 	then 
 		PREPEND="`basename $0`: ${ACTION} [${NAME2}] irq=${IRQ} pid=${lPID}"
@@ -133,7 +138,9 @@
 			echo "${PREPEND}: ERROR."
 			;;
 		esac
-		RTIRQ_TRAIL=":${IRQ}${RTIRQ_TRAIL}"
+		if [ "$loopcnt" -eq "${#PID[*]}" ];then
+			RTIRQ_TRAIL=":${IRQ}${RTIRQ_TRAIL}"
+		fi
 	fi
 	done
 }
