--- init.d/rtirq.sh	2006-09-05 10:15:37.000000000 +0200
+++ init.d/rtirq.sh	2006-09-14 21:41:59.050473196 +0200
@@ -104,13 +104,14 @@
 	fi
 	# And now the proper threading prioritization...
 	PID=`${RTIRQ_PIDOF} "IRQ ${IRQ}"`
-	if [ $((${PID})) -gt 0 -a -z "`echo ${RTIRQ_TRAIL} | grep :${IRQ}:`" ]
+	for lPID in ${PID};do
+	if [ $((${lPID})) -gt 0 -a -z "`echo ${RTIRQ_TRAIL} | grep :${IRQ}:`" ]
 	then 
-		PREPEND="`basename $0`: ${ACTION} [${NAME2}] irq=${IRQ} pid=${PID}"
+		PREPEND="`basename $0`: ${ACTION} [${NAME2}] irq=${IRQ} pid=${lPID}"
 		case ${ACTION} in
 		start)
 			PREPEND="${PREPEND} prio=${PRI2}"
-			if ${RTIRQ_CHRT} --pid --fifo ${PRI2} ${PID}
+			if ${RTIRQ_CHRT} --pid --fifo ${PRI2} ${lPID}
 			then
 				echo "${PREPEND}: OK."
 			else 
@@ -118,7 +119,7 @@
 			fi
 			;;
 		stop)
-			if ${RTIRQ_CHRT} --pid --other 0 ${PID}
+			if ${RTIRQ_CHRT} --pid --other 0 ${lPID}
 			then
 				echo "${PREPEND}: OK."
 			else 
@@ -126,7 +127,7 @@
 			fi
 			;;
 		status)
-			echo "${PREPEND}: " && ${RTIRQ_CHRT} --pid --verbose ${PID}
+			echo "${PREPEND}: " && ${RTIRQ_CHRT} --pid --verbose ${lPID}
 			;;
 		*)
 			echo "${PREPEND}: ERROR."
@@ -134,6 +135,7 @@
 		esac
 		RTIRQ_TRAIL=":${IRQ}${RTIRQ_TRAIL}"
 	fi
+	done
 }
 
 #
