diff -ru radium-1.9.1/api/api_noteedit.c radium-master/api/api_noteedit.c
--- radium-1.9.1/api/api_noteedit.c	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/api/api_noteedit.c	2012-11-12 09:08:12.000000000 +0100
@@ -69,6 +69,11 @@
 						);
 }
 
+extern int g_downscroll;
+void setNoteScrollLength(int l){
+  g_downscroll = l;
+}
+
 int getMaxVolume(){
   return MAX_VELOCITY;
 }
diff -ru radium-1.9.1/api/api_various.c radium-master/api/api_various.c
--- radium-1.9.1/api/api_various.c	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/api/api_various.c	2012-11-12 09:08:12.000000000 +0100
@@ -89,14 +89,26 @@
 }
 
 void showHideInstrumentWidget(int windownum){
-  struct Tracker_Windows *window=getWindowFromNum(windownum);if(window==NULL) return;
-  GFX_showHideInstrumentWidget(window);
+  //struct Tracker_Windows *window=getWindowFromNum(windownum);if(window==NULL) return;
+  //GFX_showHideInstrumentWidget(window);
+  if(GFX_InstrumentWindowIsVisible())
+    GFX_InstrumentWindowToBack();
+  else
+    GFX_InstrumentWindowToFront();
+}
+
+#if 0
+void toggleInstrumentWidgetOnly(void){
+  //if(GFX_
 }
+#endif
+
 void showHidePlaylist(int windownum){
   struct Tracker_Windows *window=getWindowFromNum(windownum);if(window==NULL) return;
   GFX_showHidePlaylist(window);
 }
 
+
 void addFX(int windownum){
   struct Tracker_Windows *window=getWindowFromNum(windownum);if(window==NULL) return;
   AddFXNodeLineCurrPos(window);
diff -ru radium-1.9.1/api/protos.conf radium-master/api/protos.conf
--- radium-1.9.1/api/protos.conf	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/api/protos.conf	2012-11-12 09:08:12.000000000 +0100
@@ -48,6 +48,8 @@
 
 incNoteVolume | int incvolume | int windownum ? -1
 
+setNoteScrollLength | int length
+
 
 #/*******************************************
 #  Navigating 
diff -ru radium-1.9.1/audio/Ladspa_plugins.cpp radium-master/audio/Ladspa_plugins.cpp
--- radium-1.9.1/audio/Ladspa_plugins.cpp	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/audio/Ladspa_plugins.cpp	2012-11-12 09:08:12.000000000 +0100
@@ -843,34 +843,37 @@
 }
 
 void create_ladspa_plugins(void){
+  char ladspa_path[1024];
 
 #if __linux__
-  if(getenv("LADSPA_PATH")==NULL)
+  if(getenv("LADSPA_PATH")==NULL){
     QMessageBox::information(NULL, "LADSPA_PATH is not set.", "LADSPA_PATH is not set.");
-  QDir dir(getenv("LADSPA_PATH")); //"/usr/lib64/ladspa");
+    return;
+  }
+  sprintf(ladspa_path,"%s",getenv("LADSPA_PATH"));
 #endif
 
 #if defined(FOR_WINDOWS) || defined(FOR_MACOSX)
-  QDir dir(QString(OS_get_program_path()) + OS_get_directory_separator() + "ladspa");
+  sprintf(ladspa_path,"%s",QString(QString(OS_get_program_path()) + OS_get_directory_separator() + "ladspa").ascii());
 #endif
 
-  dir.setFilter(QDir::Files | QDir::NoDotAndDotDot);
-  dir.setSorting(QDir::Name);
-  
-  QFileInfoList list = dir.entryInfoList();
-
-  for (int i = 0; i < list.size(); ++i) {
-    QFileInfo fileInfo = list.at(i);
-    if(fileInfo.suffix()==LIB_SUFFIX)
-      add_ladspa_plugin_type(fileInfo);
-  }
+  char *dirname = strtok (ladspa_path, ":");
+  do{
 
-#if 0
-  {
-    QFileInfo zitaInfo = QString(OS_get_program_path()) + "/zita_rev_ladspa.so";
-    add_ladspa_plugin_type(zitaInfo);
-  }
-#endif
+    QDir dir(dirname);
+
+    dir.setFilter(QDir::Files | QDir::NoDotAndDotDot);
+    dir.setSorting(QDir::Name);
+    
+    QFileInfoList list = dir.entryInfoList();
+    
+    for (int i = 0; i < list.size(); ++i) {
+      QFileInfo fileInfo = list.at(i);
+      if(fileInfo.suffix()==LIB_SUFFIX)
+        add_ladspa_plugin_type(fileInfo);
+    }
+
+  }while ((dirname = strtok (NULL, ":")));
 
   init_menues();
 }
diff -ru radium-1.9.1/audio/Sampler_plugin.c radium-master/audio/Sampler_plugin.c
--- radium-1.9.1/audio/Sampler_plugin.c	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/audio/Sampler_plugin.c	2012-11-12 09:08:12.000000000 +0100
@@ -81,7 +81,7 @@
   double frequency_table[128];
 } Sample;
 
-// A voice object points to only one sample. Stereo-files uses two voice objects. Soundfonts using x sounds to play a note, needs x voice objects to play that note.
+// A voice object points to only one sample. Stereo-files uses two voice objects. Soundfonts using x sounds to play a note, need x voice objects to play that note.
 typedef struct _Voice{
   struct _Voice *prev;
   struct _Voice *next;
@@ -741,14 +741,49 @@
 #include "Sampler_plugin_xi_load.c"
 #include "Sampler_plugin_sf2_load.c"
 
+static float *load_interleaved_samples(const char *filename, SF_INFO *sf_info){
+  SNDFILE *sndfile          = sf_open(filename,SFM_READ,sf_info);
+  if(sndfile==NULL)
+    return NULL;
+
+  float   *ret              = talloc_atomic(sizeof(float) * sf_info->channels * sf_info->frames);
+  int      allocated_frames = sf_info->frames;
+
+  int total_read_frames     = sf_readf_float(sndfile, ret, sf_info->frames);
+
+  if(total_read_frames==0)
+    return NULL;
+
+  while(true){
+    float samples[1024*sf_info->channels];
+    int read_now = sf_readf_float(sndfile, samples, 1024);
+    if(read_now==0)
+      break;
+
+    if(total_read_frames + read_now > allocated_frames){
+      allocated_frames = (total_read_frames+read_now) * 2;
+      ret = talloc_realloc(ret, allocated_frames * sizeof(float) * sf_info->channels);
+    }
+
+    memcpy(ret + (total_read_frames*sf_info->channels), samples, sizeof(float)*1024*sf_info->channels);
+
+    total_read_frames += read_now;
+  }
+
+  sf_close(sndfile);
+
+  sf_info->frames = total_read_frames;
+  return ret;
+}
+
 static bool load_sample_with_libsndfile(Data *data, const char *filename){
   SF_INFO sf_info; memset(&sf_info,0,sizeof(sf_info));
 
   data->num_different_samples = 1;
 
-  SNDFILE *sndfile = sf_open(filename,SFM_READ,&sf_info);
+  float *samples = load_interleaved_samples(filename, &sf_info);
 
-  if(sndfile==NULL){
+  if(samples==NULL){
     fprintf(stderr,"could not open file\n");
     return false;
   }
@@ -763,35 +798,23 @@
 
     int ch;
     for(ch=0;ch<num_channels;ch++){
-      Sample *sample=(Sample*)&data->samples[ch];
-      sample->num_frames   = sf_info.frames;
-      sample->data = malloc(sizeof(float)*sample->num_frames);
+      Sample *sample     = (Sample*)&data->samples[ch];
+      sample->num_frames = sf_info.frames;
+      sample->data       = malloc(sizeof(float)*sample->num_frames);
     }
 
-    int frames_read = 0;
-    while(frames_read < sf_info.frames){
-      float samples[1024*sf_info.channels];
-
-      int read_now = sf_readf_float(sndfile, samples, 1024);
-      //printf("Reading %d frames. Size: %d. Read: %d. samples[0]: %f. channels: %d/%d\n",read_now,sample->num_frames,frames_read,samples[0],sample->num_channels,sf_info.channels);
-
-      int interleaved_pos=0;
-      int i;
-      for(i=0;i<read_now;i++){
-        for(ch=0;ch<sf_info.channels;ch++){
-          if(ch<2){
-            Sample *sample=(Sample*)&data->samples[ch];
-            sample->data[frames_read+i] = samples[interleaved_pos];
-          }
-          interleaved_pos++;
+    int interleaved_pos=0;
+    int i;
+    for(i=0;i<sf_info.frames;i++){
+      for(ch=0;ch<sf_info.channels;ch++){
+        if(ch<2){
+          Sample *sample=(Sample*)&data->samples[ch];
+          sample->data[i] = samples[interleaved_pos];
         }
+        interleaved_pos++;
       }
-
-      frames_read += read_now;
     }
 
-    sf_close(sndfile);
-
     for(ch=0;ch<num_channels;ch++){     
       Sample *sample=(Sample*)&data->samples[ch];
 
diff -ru radium-1.9.1/bin/keybindings.conf radium-master/bin/keybindings.conf
--- radium-1.9.1/bin/keybindings.conf	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/bin/keybindings.conf	2012-11-12 09:08:12.000000000 +0100
@@ -323,16 +323,19 @@
 ####################################################################
 
 1:
-        CTRL_L		: ra.editorWindowToFront
-	ALT_R		: ra.maximizeEditorWindow
-	EXTRA_R		: ra.minimizeEditorWindow
+        CTRL_L		: ra.setNoteScrollLength 1
 
 2:
-	CTRL_L		: ra.playListWindowToFront
+        CTRL_L		: ra.setNoteScrollLength 2
+
 3:
-	CTRL_L		: ra.instrumentWindowToFront
+        CTRL_L		: ra.setNoteScrollLength 3
+
 4:
-	CTRL_L		: ra.helpWindowToFront
+        CTRL_L		: ra.setNoteScrollLength 4
+
+5:
+        CTRL_L		: ra.setNoteScrollLength 5
 
 
 BACKSPACE		: ra.insertReallines -1
diff -ru radium-1.9.1/bin/menues.conf radium-master/bin/menues.conf
--- radium-1.9.1/bin/menues.conf	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/bin/menues.conf	2012-11-12 09:08:12.000000000 +0100
@@ -74,6 +74,12 @@
 		Add               SHIFT_L+<key>
 		Delete                    Delete
 		Stop Notes                Delete
+		---------------
+		Set Note Scroll Length 1 | ra.setNoteScrollLength(1)
+		Set Note Scroll Length 2 | ra.setNoteScrollLength(2)
+		Set Note Scroll Length 3 | ra.setNoteScrollLength(3)
+		Set Note Scroll Length 4 | ra.setNoteScrollLength(4)
+		Set Note Scroll Length 5 | ra.setNoteScrollLength(5)
 	Velocities
 		Del 			| ra.generalDelete()
 		Def Pos 		| ra.generalReturn()
diff -ru radium-1.9.1/bin/new_song.rad radium-master/bin/new_song.rad
--- radium-1.9.1/bin/new_song.rad	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/bin/new_song.rad	2012-11-12 09:08:12.000000000 +0100
@@ -53,10 +53,10 @@
 0
 
 ?width
-1062
+731
 
 ?height
-693
+460
 
 ?fontID
 0
@@ -68,10 +68,10 @@
 10
 
 ?fontheight
-18
+19
 
 ?org_fontheight
-18
+19
 
 ?curr_block
 0
@@ -125,16 +125,16 @@
 30
 
 ?temponodearea
-156
+70
 
 ?maxwtracksize
 0
 
 ?curr_realline
-18
+62
 
 ?num_reallines
-100
+64
 
 ?reltempomax
 2.000000
@@ -156,7 +156,7 @@
 
 \
 LOCALZOOMS
-100
+64
 /
 
 \
@@ -308,7 +308,7 @@
 8
 
 ?num_lines
-100
+64
 
 ?reltempo
 1.001876
@@ -549,7 +549,7 @@
 RELTEMPO
 0.000000
 0.000000
-99
+63
 65533
 65534
 /
@@ -565,6 +565,13 @@
 INSTRUMENT
 
 ?instrumentname
+MIDI instrument
+/
+
+\
+INSTRUMENT
+
+?instrumentname
 Audio instrument
 
 \
@@ -1511,10 +1518,10 @@
 7
 from_patch
 INT_TYPE
-17
+0
 to_patch
 INT_TYPE
-20
+1
 << HASH MAP END
 <int hash> 4
 HASH_TYPE
@@ -1522,10 +1529,10 @@
 7
 from_patch
 INT_TYPE
-20
+19
 to_patch
 INT_TYPE
-0
+7
 << HASH MAP END
 <int hash> 0
 HASH_TYPE
@@ -1533,10 +1540,10 @@
 7
 from_patch
 INT_TYPE
-19
+20
 to_patch
 INT_TYPE
-7
+0
 << HASH MAP END
 <int hash> 1
 HASH_TYPE
@@ -1544,10 +1551,10 @@
 7
 from_patch
 INT_TYPE
-0
+17
 to_patch
 INT_TYPE
-1
+20
 << HASH MAP END
 num_connections
 INT_TYPE
@@ -1567,115 +1574,130 @@
 11
 name
 STRING_TYPE
-System Out
+Zita Reverb
 type_name
 STRING_TYPE
-Jack
+Faust
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
-2
-System Width On/Off
+11
+System Eq2 On/Off
 FLOAT_TYPE
-0.000000
-System Hs L.
+1.000000
+System Eq2 L.
 FLOAT_TYPE
-0.0
-System Hs F.
+-5.243000
+System Eq1 F.
 FLOAT_TYPE
-4500
-System Ls F.
+399.359863
+System Out
 FLOAT_TYPE
-400
-System Eq2 F.
+0.999712
+Low RT60
 FLOAT_TYPE
-1000
-System Eq1 F.
+3.587200
+LF X
 FLOAT_TYPE
-400
-System Lowpass On/Off
+199.910004
+System Hs On/Off
+FLOAT_TYPE
+1.000000
+System Ls On/Off
 FLOAT_TYPE
 0.000000
-System Dry/Wet
+System Ls L.
 FLOAT_TYPE
-1.000000
-System Pan On/Off
+0.000000
+System Chorus
 FLOAT_TYPE
 0.000000
-System Reverb On/Off
+Mid RT60
+FLOAT_TYPE
+4.755500
+Input: In Delay
+FLOAT_TYPE
+0.000000
+System Effects On/Off
 FLOAT_TYPE
 1.000000
 System Reverb
 FLOAT_TYPE
 0.000000
-System Out
+System Dry/Wet
+FLOAT_TYPE
+1.000000
+System Ls F.
+FLOAT_TYPE
+399.359863
+System Eq2 F.
+FLOAT_TYPE
+1000.000366
+System Lp F.
 FLOAT_TYPE
-1.0
+4999.347168
 System Volume On/Off
 FLOAT_TYPE
 1.000000
-System Volume
-FLOAT_TYPE
-0.0
 System In On/Off
 FLOAT_TYPE
 1.000000
-System In
-FLOAT_TYPE
-1.0
-System Width
+System Pan
 FLOAT_TYPE
-0.005000
-System Hs On/Off
+0.500000
+System Reverb On/Off
 FLOAT_TYPE
-0.000000
-System Ls On/Off
+1.000000
+System Pan On/Off
 FLOAT_TYPE
 0.000000
-System Ls L.
+System Out On/Off
 FLOAT_TYPE
-0.0
-System Eq2 On/Off
+1.000000
+System Hs F.
 FLOAT_TYPE
-0.000000
-System Eq2 L.
+4499.011719
+System Chorus On/Off
 FLOAT_TYPE
-0.0
-System Eq1 On/Off
+1.000000
+System Eq1 L.
 FLOAT_TYPE
 0.000000
-System Eq1 L.
+System In
 FLOAT_TYPE
-0.0
-System Lp F.
+0.999712
+HF Damping
 FLOAT_TYPE
-5000
-System Effects On/Off
+3657.959961
+System Width On/Off
 FLOAT_TYPE
-1.000000
-System Pan
+0.000000
+System Width
 FLOAT_TYPE
-0.500000
-System Chorus On/Off
+0.190000
+System Hs L.
 FLOAT_TYPE
-1.000000
-System Chorus
+9.639000
+System Eq1 On/Off
 FLOAT_TYPE
 0.000000
-System Out On/Off
+System Lowpass On/Off
 FLOAT_TYPE
-1.000000
+0.000000
+System Volume
+FLOAT_TYPE
+1.039322
 << HASH MAP END
 << HASH MAP END
 patch
 INT_TYPE
-1
+7
 y
 FLOAT_TYPE
--35.000000
+-75.000000
 x
 FLOAT_TYPE
-120.000000
+-120.000000
 << HASH MAP END
 <int hash> 1
 HASH_TYPE
@@ -1687,10 +1709,10 @@
 11
 name
 STRING_TYPE
-Bus 1
+Pipe
 type_name
 STRING_TYPE
-Bus
+Pipe
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
@@ -1703,16 +1725,16 @@
 0.000000
 System Hs F.
 FLOAT_TYPE
-1500
+4499.011719
 System Ls F.
 FLOAT_TYPE
-400
+399.359863
 System Eq2 F.
 FLOAT_TYPE
-1000
+1000.000366
 System Eq1 F.
 FLOAT_TYPE
-400
+399.359863
 System Lowpass On/Off
 FLOAT_TYPE
 0.000000
@@ -1730,7 +1752,7 @@
 0.000000
 System Out
 FLOAT_TYPE
-1.0
+0.999712
 System Volume On/Off
 FLOAT_TYPE
 1.000000
@@ -1742,10 +1764,10 @@
 1.000000
 System In
 FLOAT_TYPE
-1.0
+0.999712
 System Width
 FLOAT_TYPE
-0.000000
+0.005000
 System Hs On/Off
 FLOAT_TYPE
 0.000000
@@ -1769,7 +1791,7 @@
 0.000000
 System Lp F.
 FLOAT_TYPE
-5000
+4999.347168
 System Effects On/Off
 FLOAT_TYPE
 1.000000
@@ -1789,13 +1811,13 @@
 << HASH MAP END
 patch
 INT_TYPE
-19
+0
 y
 FLOAT_TYPE
--75.000000
+-35.000000
 x
 FLOAT_TYPE
--240.000000
+0.000000
 << HASH MAP END
 <int hash> 4
 HASH_TYPE
@@ -1807,10 +1829,10 @@
 11
 name
 STRING_TYPE
-Pipe
+Bus 1
 type_name
 STRING_TYPE
-Pipe
+Bus
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
@@ -1820,19 +1842,19 @@
 0.000000
 System Hs L.
 FLOAT_TYPE
-0.0
+0.000000
 System Hs F.
 FLOAT_TYPE
-4500
+1499.337646
 System Ls F.
 FLOAT_TYPE
-400
+399.359863
 System Eq2 F.
 FLOAT_TYPE
-1000
+1000.000366
 System Eq1 F.
 FLOAT_TYPE
-400
+399.359863
 System Lowpass On/Off
 FLOAT_TYPE
 0.000000
@@ -1850,22 +1872,22 @@
 0.000000
 System Out
 FLOAT_TYPE
-1.0
+0.999712
 System Volume On/Off
 FLOAT_TYPE
 1.000000
 System Volume
 FLOAT_TYPE
-1.0
+0.999712
 System In On/Off
 FLOAT_TYPE
 1.000000
 System In
 FLOAT_TYPE
-1.0
+0.999712
 System Width
 FLOAT_TYPE
-0.005000
+0.000000
 System Hs On/Off
 FLOAT_TYPE
 0.000000
@@ -1874,22 +1896,22 @@
 0.000000
 System Ls L.
 FLOAT_TYPE
-0.0
+0.000000
 System Eq2 On/Off
 FLOAT_TYPE
 0.000000
 System Eq2 L.
 FLOAT_TYPE
-0.0
+0.000000
 System Eq1 On/Off
 FLOAT_TYPE
 0.000000
 System Eq1 L.
 FLOAT_TYPE
-0.0
+0.000000
 System Lp F.
 FLOAT_TYPE
-5000
+4999.347168
 System Effects On/Off
 FLOAT_TYPE
 1.000000
@@ -1909,13 +1931,13 @@
 << HASH MAP END
 patch
 INT_TYPE
-0
+19
 y
 FLOAT_TYPE
--35.000000
+-75.000000
 x
 FLOAT_TYPE
-0.000000
+-240.000000
 << HASH MAP END
 num_chips
 INT_TYPE
@@ -1930,148 +1952,115 @@
 11
 name
 STRING_TYPE
-Calf MultiChorus LADSPA
+Bus 2
 type_name
 STRING_TYPE
-Ladspa
+Bus
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
-17
-Dry Amount
+2
+System Width On/Off
 FLOAT_TYPE
 0.000000
-System Ls On/Off
+System Hs L.
 FLOAT_TYPE
 0.000000
-System Dry/Wet
+System Hs F.
 FLOAT_TYPE
-1.000000
-Center Frq 2
+1499.337646
+System Ls F.
 FLOAT_TYPE
-4999.503906
-System Width On/Off
+399.359863
+System Eq2 F.
 FLOAT_TYPE
-0.000000
-System Hs L.
+1000.000366
+System Eq1 F.
 FLOAT_TYPE
-0.000000
+399.359863
 System Lowpass On/Off
 FLOAT_TYPE
 0.000000
-Minimum delay
-FLOAT_TYPE
-4.999510
-Modulation depth
+System Dry/Wet
 FLOAT_TYPE
-5.999410
-System Ls F.
+1.000000
+System Pan On/Off
 FLOAT_TYPE
-400
-System Pan
+0.000000
+System Reverb On/Off
 FLOAT_TYPE
-0.500000
+1.000000
 System Reverb
 FLOAT_TYPE
 0.000000
-System Out On/Off
+System Out
 FLOAT_TYPE
-1.000000
+0.999712
 System Volume On/Off
 FLOAT_TYPE
 1.000000
-Center Frq 1
-FLOAT_TYPE
-97.956001
-Voices
-FLOAT_TYPE
-3.999500
-System Eq2 On/Off
-FLOAT_TYPE
-0.000000
-System Eq1 F.
+System Volume
 FLOAT_TYPE
-400
-System Effects On/Off
+0.999712
+System In On/Off
 FLOAT_TYPE
 1.000000
-System Chorus On/Off
+System In
 FLOAT_TYPE
-1.000000
-System Chorus
+0.999712
+System Width
 FLOAT_TYPE
 0.000000
-Modulation rate
-FLOAT_TYPE
-0.499755
-System Pan On/Off
+System Hs On/Off
 FLOAT_TYPE
 0.000000
-System Width
+System Ls On/Off
 FLOAT_TYPE
 0.000000
-System Eq1 On/Off
+System Ls L.
 FLOAT_TYPE
 0.000000
-Amount
+System Eq2 On/Off
 FLOAT_TYPE
-2.000000
+0.000000
 System Eq2 L.
 FLOAT_TYPE
 0.000000
-System Eq2 F.
+System Eq1 On/Off
 FLOAT_TYPE
-1000
+0.000000
 System Eq1 L.
 FLOAT_TYPE
 0.000000
-System Ls L.
+System Lp F.
 FLOAT_TYPE
-0.000000
-System Reverb On/Off
+1499.337646
+System Effects On/Off
 FLOAT_TYPE
 1.000000
-System In On/Off
+System Pan
+FLOAT_TYPE
+0.500000
+System Chorus On/Off
 FLOAT_TYPE
 1.000000
-System Hs On/Off
+System Chorus
 FLOAT_TYPE
 0.000000
-Inter-voice phase
-FLOAT_TYPE
-63.972000
-System Hs F.
-FLOAT_TYPE
-1500
-System Lp F.
-FLOAT_TYPE
-5000
-System Out
-FLOAT_TYPE
-1.0
-Q
-FLOAT_TYPE
-0.125000
-Stereo phase
-FLOAT_TYPE
-180.000000
-System In
-FLOAT_TYPE
-1.0
-System Volume
+System Out On/Off
 FLOAT_TYPE
-0.999712
+1.000000
 << HASH MAP END
 << HASH MAP END
 patch
 INT_TYPE
-20
+17
 y
 FLOAT_TYPE
 -115.000000
 x
 FLOAT_TYPE
--120.000000
+-240.000000
 << HASH MAP END
 <int hash> 2
 HASH_TYPE
@@ -2083,130 +2072,115 @@
 11
 name
 STRING_TYPE
-Zita Reverb
+System Out
 type_name
 STRING_TYPE
-Faust
+Jack
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
-11
-System Eq2 On/Off
-FLOAT_TYPE
-1.000000
-System Eq2 L.
-FLOAT_TYPE
--5.243000
-System Eq1 F.
-FLOAT_TYPE
-400
-System Out
+2
+System Width On/Off
 FLOAT_TYPE
-1.0
-Low RT60
+0.000000
+System Hs L.
 FLOAT_TYPE
-3.587200
-LF X
+0.000000
+System Hs F.
 FLOAT_TYPE
-199.910004
-System Hs On/Off
+4499.011719
+System Ls F.
 FLOAT_TYPE
-1.000000
-System Ls On/Off
+399.359863
+System Eq2 F.
 FLOAT_TYPE
-0.000000
-System Ls L.
+1000.000366
+System Eq1 F.
 FLOAT_TYPE
-0.000000
-System Chorus
+399.359863
+System Lowpass On/Off
 FLOAT_TYPE
 0.000000
-Mid RT60
+System Dry/Wet
 FLOAT_TYPE
-4.755500
-Input: In Delay
+1.000000
+System Pan On/Off
 FLOAT_TYPE
 0.000000
-System Effects On/Off
+System Reverb On/Off
 FLOAT_TYPE
 1.000000
 System Reverb
 FLOAT_TYPE
 0.000000
-System Dry/Wet
-FLOAT_TYPE
-1.000000
-System Ls F.
-FLOAT_TYPE
-400
-System Eq2 F.
-FLOAT_TYPE
-1000
-System Lp F.
+System Out
 FLOAT_TYPE
-5000
+0.999712
 System Volume On/Off
 FLOAT_TYPE
 1.000000
+System Volume
+FLOAT_TYPE
+0.000000
 System In On/Off
 FLOAT_TYPE
 1.000000
-System Pan
+System In
 FLOAT_TYPE
-0.500000
-System Reverb On/Off
+0.999712
+System Width
 FLOAT_TYPE
-1.000000
-System Pan On/Off
+0.005000
+System Hs On/Off
 FLOAT_TYPE
 0.000000
-System Out On/Off
-FLOAT_TYPE
-1.000000
-System Hs F.
+System Ls On/Off
 FLOAT_TYPE
-4500
-System Chorus On/Off
+0.000000
+System Ls L.
 FLOAT_TYPE
-1.000000
-System Eq1 L.
+0.000000
+System Eq2 On/Off
 FLOAT_TYPE
 0.000000
-System In
+System Eq2 L.
 FLOAT_TYPE
-1.0
-HF Damping
+0.000000
+System Eq1 On/Off
 FLOAT_TYPE
-3657.959961
-System Width On/Off
+0.000000
+System Eq1 L.
 FLOAT_TYPE
 0.000000
-System Width
+System Lp F.
 FLOAT_TYPE
-0.190000
-System Hs L.
+4999.347168
+System Effects On/Off
 FLOAT_TYPE
-9.639000
-System Eq1 On/Off
+1.000000
+System Pan
 FLOAT_TYPE
-0.000000
-System Lowpass On/Off
+0.500000
+System Chorus On/Off
+FLOAT_TYPE
+1.000000
+System Chorus
 FLOAT_TYPE
 0.000000
-System Volume
+System Out On/Off
 FLOAT_TYPE
-1.039322
+1.000000
 << HASH MAP END
 << HASH MAP END
 patch
 INT_TYPE
-7
+1
 y
 FLOAT_TYPE
--75.000000
+-35.000000
 x
 FLOAT_TYPE
--120.000000
+120.000000
 << HASH MAP END
 <int hash> 0
 HASH_TYPE
@@ -2218,115 +2192,148 @@
 11
 name
 STRING_TYPE
-Bus 2
+Calf MultiChorus LADSPA
 type_name
 STRING_TYPE
-Bus
+Ladspa
 effects
 HASH_TYPE
 >> HASH MAP BEGIN
-2
-System Width On/Off
+17
+Dry Amount
 FLOAT_TYPE
 0.000000
-System Hs L.
+System Ls On/Off
 FLOAT_TYPE
 0.000000
-System Hs F.
+System Dry/Wet
 FLOAT_TYPE
-1500
-System Ls F.
+1.000000
+Center Frq 2
 FLOAT_TYPE
-400
-System Eq2 F.
+4999.450195
+System Width On/Off
 FLOAT_TYPE
-1000.000366
-System Eq1 F.
+0.000000
+System Hs L.
 FLOAT_TYPE
-400
+0.000000
 System Lowpass On/Off
 FLOAT_TYPE
 0.000000
-System Dry/Wet
+Minimum delay
 FLOAT_TYPE
-1.000000
-System Pan On/Off
+4.998043
+Modulation depth
 FLOAT_TYPE
-0.000000
-System Reverb On/Off
+5.997910
+System Ls F.
 FLOAT_TYPE
-1.000000
+399.359863
+System Pan
+FLOAT_TYPE
+0.500000
 System Reverb
 FLOAT_TYPE
 0.000000
-System Out
+System Out On/Off
 FLOAT_TYPE
-1.0
+1.000000
 System Volume On/Off
 FLOAT_TYPE
 1.000000
-System Volume
+Center Frq 1
 FLOAT_TYPE
-0.999712
-System In On/Off
+97.867630
+Voices
+FLOAT_TYPE
+3.999500
+System Eq2 On/Off
+FLOAT_TYPE
+0.000000
+System Eq1 F.
+FLOAT_TYPE
+399.359863
+System Effects On/Off
 FLOAT_TYPE
 1.000000
-System In
+System Chorus On/Off
 FLOAT_TYPE
-1.0
-System Width
+1.000000
+System Chorus
 FLOAT_TYPE
 0.000000
-System Hs On/Off
+Modulation rate
 FLOAT_TYPE
-0.000000
-System Ls On/Off
+0.499701
+System Pan On/Off
 FLOAT_TYPE
 0.000000
-System Ls L.
+System Width
 FLOAT_TYPE
 0.000000
-System Eq2 On/Off
+System Eq1 On/Off
 FLOAT_TYPE
 0.000000
+Amount
+FLOAT_TYPE
+2.000000
 System Eq2 L.
 FLOAT_TYPE
 0.000000
-System Eq1 On/Off
+System Eq2 F.
 FLOAT_TYPE
-0.000000
+1000.000366
 System Eq1 L.
 FLOAT_TYPE
 0.000000
-System Lp F.
+System Ls L.
 FLOAT_TYPE
-1500
-System Effects On/Off
+0.000000
+System Reverb On/Off
 FLOAT_TYPE
 1.000000
-System Pan
-FLOAT_TYPE
-0.500000
-System Chorus On/Off
+System In On/Off
 FLOAT_TYPE
 1.000000
-System Chorus
+System Hs On/Off
 FLOAT_TYPE
 0.000000
-System Out On/Off
+Inter-voice phase
 FLOAT_TYPE
-1.000000
+63.972000
+System Hs F.
+FLOAT_TYPE
+1499.337646
+System Lp F.
+FLOAT_TYPE
+4999.347168
+System Out
+FLOAT_TYPE
+0.999712
+Q
+FLOAT_TYPE
+0.125000
+Stereo phase
+FLOAT_TYPE
+180.000000
+System In
+FLOAT_TYPE
+0.999712
+System Volume
+FLOAT_TYPE
+0.999712
 << HASH MAP END
 << HASH MAP END
 patch
 INT_TYPE
-17
+20
 y
 FLOAT_TYPE
 -115.000000
 x
 FLOAT_TYPE
--240.000000
+-120.000000
 << HASH MAP END
 << HASH MAP END
 << HASH MAP END
diff -ru radium-1.9.1/bin/packages/build.sh radium-master/bin/packages/build.sh
--- radium-1.9.1/bin/packages/build.sh	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/bin/packages/build.sh	2012-11-12 09:08:12.000000000 +0100
@@ -69,7 +69,7 @@
 tar xvzf fluidsynth-1.1.6.tar.gz
 cd fluidsynth-1.1.6
 make clean
-CFLAGS="-mtune=native -O3" CPPFLAGS="-mtune=native -O3" ./configure --enable-static --disable-aufile-support --disable-pulse-support --disable-alsa-support --disable-libsndfile-support --disable-portaudio-support --disable-oss-support --disable-midishare --disable-jack-support --disable-coreaudio --disable-coremidi --disable-dart --disable-lash --disable-ladcca --disable-aufile-support --disable-dbus-support
+CFLAGS="-mtune=native -O3" CPPFLAGS="-mtune=native -O3" ./configure --enable-static --disable-aufile-support --disable-pulse-support --disable-alsa-support --disable-libsndfile-support --disable-portaudio-support --disable-oss-support --disable-midishare --disable-jack-support --disable-coreaudio --disable-coremidi --disable-dart --disable-lash --disable-ladcca --disable-aufile-support --disable-dbus-support --without-readline
 # --enable-debug
 make -j3
 cd ..
diff -ru radium-1.9.1/build_linux_binary.sh radium-master/build_linux_binary.sh
--- radium-1.9.1/build_linux_binary.sh	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/build_linux_binary.sh	2012-11-12 09:08:12.000000000 +0100
@@ -4,8 +4,14 @@
 
 export IS_LINUX_BINARY="-DIS_LINUX_BINARY"
 
+rm -f Qt_Main.o
+
+export OPTIMIZE="-mtune=generic -msse -O3"
+export BUILDTYPE=RELEASE
+
 ./build_linux.sh $@
 
+
 cp README radium64_dist/
 cp bin/radium radium64_dist/bin/
 cp bin/*.py radium64_dist/bin/
@@ -14,7 +20,7 @@
 cp bin/colors radium64_dist/bin/
 cp bin/menues.conf radium64_dist/bin/
 cp bin/keybindings.conf radium64_dist/bin/
-rm -f radium_0.9.1_linux64.tar.gz && tar cvzf radium_0.9.1_linux64.tar.gz radium64_dist
+rm -f radium_1.9.1_linux64.tar.gz && tar cvzf radium_1.9.1_linux64.tar.gz radium64_dist
 
 
 #echo "Building finished."
diff -ru radium-1.9.1/build_linux_common.sh radium-master/build_linux_common.sh
--- radium-1.9.1/build_linux_common.sh	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/build_linux_common.sh	2012-11-12 09:08:12.000000000 +0100
@@ -7,8 +7,14 @@
 export UIC="`./find_moc_and_uic_paths.sh uic`"
 
 #export BUILDTYPE=RELEASE
-export BUILDTYPE=DEBUG
-export OPTIMIZE="-mtune=native -O3"
+if ! env |grep BUILDTYPE ; then
+    export BUILDTYPE=DEBUG
+fi
+
+if ! env |grep OPTIMIZE ; then
+    export OPTIMIZE="-mtune=native -O3"
+fi
+
 export CPUOPT="$OPTIMIZE" # Some files are so CPU intensive that we need to turn on optimizations even in debug mode, at least when running in valgrind.
 
 export CCC=g++
diff -ru radium-1.9.1/Changelog radium-master/Changelog
--- radium-1.9.1/Changelog	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Changelog	2012-11-12 09:08:12.000000000 +0100
@@ -1,3 +1,37 @@
+
+Most important changes 1.9.1 -> 1.9.2:
+* Give editor keyboard focus when clicking it
+* Option to set number of lines to scroll when adding notes. (suggested by Florian Paul Schmidt)
+* Right click to reset slider value
+* Properly load samples. (Thanks to Erik de Castro Lopo for explaining the libsndfile API)
+* Avoid playing samples two times (almost exactly on top of each other) when clicking the filename of a sample. (Bug found by Florian Paul Schmidt)
+* Make bindir and libdir overridable in the Makefile. Requested by Dominique Michel to easier make ebuilds for Radium.
+* Support several paths in LADSPA_PATH (Bug reported by Florian Paul Schmidt)
+* Fix readline build errors (Bug reported by Florian Paul Schmidt)
+
+
+
+Most important changes 0.69.3 -> 1.9.1:
+* Audio.
+* A quick-to-use modular mixer. Jack ports can be inserted anywhere.
+* VST plugins and instruments (native, not via wine)
+* LADSPA
+* Fluidsynth instrument
+* 20 STK instruments. (Physical modelling)
+* Sampler instrument (soundfiles, xi instruments, soundfonts)
+* Multiband compressor
+* Built-in filters and equalizers.
+* Simple arpeggiator
+* Drunk input velocity
+* Various GUI improvements
+* Pitch expand made available in the menues.
+* Tapiir plugin
+* Zita reverb plugin
+* Demo song
+* Reconfigured keybindings
+* Various bug fixes
+
+
 Most important changes 0.69.2 -> 0.69.3:
 * Fix font update problem after changing size
 * Add Append Track function. Using F key, same as Append Block.
diff -ru radium-1.9.1/common/notes.c radium-master/common/notes.c
--- radium-1.9.1/common/notes.c	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/common/notes.c	2012-11-12 09:08:12.000000000 +0100
@@ -201,6 +201,8 @@
   return new_velocity;
 }
 
+int g_downscroll = 1;
+
 void InsertNoteCurrPos(struct Tracker_Windows *window,int notenum, int override){
 	struct WBlocks *wblock;
 	struct WTracks *wtrack;
@@ -244,7 +246,7 @@
 					    
 		}
 		if(window->curr_track_sub==-1 && !pc->isplaying){
-			ScrollEditorDown(window,1);
+			ScrollEditorDown(window,g_downscroll);
 		}
 
 	}else{
@@ -277,7 +279,7 @@
 			UpdateAllWTracksCoordinates(window,wblock);
 
 		if(window->curr_track_sub==-1 && override==0 && !pc->isplaying){
-			ScrollEditorDown(window,1);
+			ScrollEditorDown(window,g_downscroll);
 		}
 
 		UpdateTrackReallines(window,wblock,wtrack);
@@ -386,7 +388,7 @@
 	}
 
 	if(window->curr_track_sub==-1  && !pc->isplaying){
-		ScrollEditorDown(window,1);
+		ScrollEditorDown(window,g_downscroll);
 	}
 	UpdateTrackReallines(window,wblock,wtrack);
 	ClearTrack(window,wblock,wtrack,wblock->top_realline,wblock->bot_realline);
diff -ru radium-1.9.1/common/visual_proc.h radium-master/common/visual_proc.h
--- radium-1.9.1/common/visual_proc.h	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/common/visual_proc.h	2012-11-12 09:08:12.000000000 +0100
@@ -82,6 +82,8 @@
 extern LANGSPEC void GFX_showHideEditor(void);
 extern LANGSPEC void GFX_showHideMixerWidget(void);
 
+extern LANGSPEC bool GFX_InstrumentWindowIsVisible(void);
+
 extern LANGSPEC void GFX_MaximizeEditorWindow(struct Tracker_Windows *tvisual);
 extern LANGSPEC void GFX_MinimizeEditorWindow(struct Tracker_Windows *tvisual);
 
diff -ru radium-1.9.1/Makefile.Qt radium-master/Makefile.Qt
--- radium-1.9.1/Makefile.Qt	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Makefile.Qt	2012-11-12 20:03:20.000000000 +0100
@@ -8,8 +8,8 @@
 # 1. Where to install
 #
 PREFIX ?= /usr/local
-bindir = $(PREFIX)/bin
-libdir = $(PREFIX)/lib
+bindir ?= $(PREFIX)/bin
+libdir ?= $(PREFIX)/lib
 
 
 # 2. Where moc and uic for QT4 is installed.
@@ -85,7 +85,7 @@
 GTK_LDFLAGS ?= `$(PKG) --libs gtk+-2.0`
 
 
-DEFINES= -DNOPAUSEPLAY
+DEFINES=-DNOPAUSEPLAY
 # -DMEMORY_DEBUG
 # define SYSBASEDEBUG
 # define TRACKER_DEBUG
@@ -264,10 +264,12 @@
 	mkdir $(DESTDIR)$(libdir)/radium/packages/xmessage-1.0.3/
 	cp bin/packages/xmessage-1.0.3/xmessage $(DESTDIR)$(libdir)/radium/packages/xmessage-1.0.3/
 
+	rm -fr $(DESTDIR)$(libdir)/radium/python2.7
+
 	chmod -R a+rx $(DESTDIR)$(libdir)/radium/*
 
 	echo "#!/bin/sh" >$(DESTDIR)$(bindir)/radium
-	echo "cd " $(DESTDIR)$(libdir)/radium >>$(DESTDIR)$(bindir)/radium
+	echo "cd " $(libdir)/radium >>$(DESTDIR)$(bindir)/radium
 	echo "./radium" >>$(DESTDIR)$(bindir)/radium
 
 	chmod a+rx $(DESTDIR)$(bindir)/radium
Seulement dans radium-master: Makefile.Qt_radium_script.patch
diff -ru radium-1.9.1/Qt/Qt_EventReceiver.cpp radium-master/Qt/Qt_EventReceiver.cpp
--- radium-1.9.1/Qt/Qt_EventReceiver.cpp	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Qt/Qt_EventReceiver.cpp	2012-11-12 09:08:12.000000000 +0100
@@ -294,6 +294,8 @@
 
   EventReciever(&tevent,this->window);
 
+  setFocus();
+
   updateEditor();
 }
 
diff -ru radium-1.9.1/Qt/Qt_instruments.cpp radium-master/Qt/Qt_instruments.cpp
--- radium-1.9.1/Qt/Qt_instruments.cpp	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Qt/Qt_instruments.cpp	2012-11-12 09:08:12.000000000 +0100
@@ -441,6 +441,7 @@
   return instruments_widget;
 }
 
+#if 0
 static void set_widget_height(int height){
   QMainWindow *main_window = static_cast<QMainWindow*>(root->song->tracker_windows->os_visual.main_window);
   EditorWidget *editor = static_cast<EditorWidget*>(root->song->tracker_windows->os_visual.widget);
@@ -458,13 +459,20 @@
 #  undef QValueList
 #endif
 }
+#endif
+
+bool GFX_InstrumentWindowIsVisible(void){
+  return instruments_widget->isVisible();
+}
 
 void GFX_InstrumentWindowToFront(void){
-  set_widget_height(300);
+  //set_widget_height(30);
+  instruments_widget->show();
 }
 
 void GFX_InstrumentWindowToBack(void){
-  set_widget_height(0);
+  instruments_widget->hide();
+  //set_widget_height(0);
 }
 
 
diff -ru radium-1.9.1/Qt/Qt_Main.cpp radium-master/Qt/Qt_Main.cpp
--- radium-1.9.1/Qt/Qt_Main.cpp	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Qt/Qt_Main.cpp	2012-11-12 09:08:12.000000000 +0100
@@ -353,7 +353,7 @@
 #if 1
     if(override_default_qt_style){
       //QApplication::setStyle( new QOxygenStyle());
-      QApplication::setStyle( new QPlastiqueStyle());
+    QApplication::setStyle( new QPlastiqueStyle());
     //QApplication::setStyle( new QCleanlooksStyle() );
     //QApplication::setStyle( new QWindowsStyle() );
     }
@@ -544,9 +544,12 @@
     //exit(0);
   }
 
-  // Create application here in order to get default style. (not recommended, but can't find another way)
-  qapplication=new MyApplication(argc,argv);
-  g_qapplication = qapplication;
+  QApplication::setDesktopSettingsAware(false);
+
+    // Create application here in order to get default style. (not recommended, but can't find another way)
+    qapplication=new MyApplication(argc,argv);
+    g_qapplication = qapplication;
+
 
   {
 #if 0
@@ -572,22 +575,27 @@
 
     QFontDatabase::addApplicationFont("fonts/LiberationMono-Bold.ttf");
     QFontDatabase::addApplicationFont("fonts/VeraMono.ttf");
+    QFontDatabase::addApplicationFont("fonts/NimbusSansL.ttf");
+
 
     //QApplication::setFont(QFont("Lohit-Tamil",8));
     //QApplication::setFont(QFont("Nimbus Sans L",8));
     //QApplication::setFont(QFont("Liberation Sans L",8));
 
-    printf("System font name: \"%s\". Size: %d\n",QApplication::font().family().ascii(),QApplication::font().pointSize());
+    //printf("System font name: \"%s\". Size: %d\n",QApplication::font().family().ascii(),QApplication::font().pointSize());
 
     int system_font_size = SETTINGS_read_int((char*)"system_font_size",-1);
     if(system_font_size>=0){
       //QFont font=QFont(QApplication::font().family(),system_font_size);
       QFont font=QFont("Nimbus Sans L",system_font_size);
+      //QFont font=QFont("Nimbus Sans L",10);
+      //font.setPointSize(system_font_size);
       //QFont font=QFont("Bitstream Vera Sans Mono",system_font_size);
       QApplication::setFont(font);
     }
   }
 
+
   signal(SIGINT,finish);
 
 #if GTK_IS_USED
diff -ru radium-1.9.1/Qt/Qt_Menues.cpp radium-master/Qt/Qt_Menues.cpp
--- radium-1.9.1/Qt/Qt_Menues.cpp	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Qt/Qt_Menues.cpp	2012-11-12 09:08:12.000000000 +0100
@@ -127,7 +127,7 @@
   //QFont sansFont("WenQuanYi Zen Hei Mono",8);
   //QFont sansFont("Aurulent Sans Mono",8);
   QFont sansFont("Bitstream Vera Sans Mono",8);
-
+  //sansFont.setPointSize(8);
   //sansFont.setBold(true);
 
   menu->menu->setFont(sansFont);
diff -ru radium-1.9.1/Qt/Qt_sample_requester_widget_callbacks.h radium-master/Qt/Qt_sample_requester_widget_callbacks.h
--- radium-1.9.1/Qt/Qt_sample_requester_widget_callbacks.h	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/Qt/Qt_sample_requester_widget_callbacks.h	2012-11-12 09:08:12.000000000 +0100
@@ -285,6 +285,7 @@
       successfully_selected = FLUIDSYNTH_set_new_preset(plugin, _sf2_file, bank_num, preset_num);
 
     if(successfully_selected==true && pc->isplaying==false)
+      printf("playing note 1\n");
       PATCH_play_note(g_currpatch, 12*_preview_octave, MAX_VELOCITY/2, NULL);
     }
 
@@ -345,6 +346,7 @@
 
     if(SAMPLER_set_new_sample(plugin,filename,file_list->currentRow()-1)==true){
       if(pc->isplaying==false)
+        printf("playing note 2\n");
         PATCH_play_note(g_currpatch, 12*_preview_octave, MAX_VELOCITY/2, NULL);
     }
   }
@@ -390,6 +392,7 @@
   }
 
   void handle_item_pressed(QString item_text){
+    printf("item pressed\n");
     switch(_file_chooser_state){
     case IN_SF2_BANK:
       handle_sf2_preset_pressed(item_text);
@@ -430,7 +433,8 @@
 
   void on_file_list_itemActivated ( QListWidgetItem * item ){
     bool was_normal_file = item!=NULL && !item->text().endsWith("/");
-    on_file_list_itemPressed(item);
+    //on_file_list_itemPressed(item);
+    handle_item_pressed(item->text());
     if(was_normal_file){
 
       if(g_filenames_hash.contains(item->text())==true)
@@ -457,11 +461,12 @@
   void on_file_list_currentItemChanged ( QListWidgetItem * current, QListWidgetItem * previous ){
     printf("Current item changed!\n");
     if( current!=NULL && !current->text().endsWith("/"))
-      on_file_list_itemPressed(current);
+      handle_item_pressed(current->text());
   }
 
   void on_file_list_itemPressed(QListWidgetItem * item ){
-    handle_item_pressed(item->text());
+    printf("file_list item pressed\n");
+    //handle_item_pressed(item->text());
   }
 
 #if 0
@@ -530,6 +535,7 @@
       
       if(SAMPLER_set_new_sample(plugin,filename,file_list->currentRow()-1)==true){
         if(pc->isplaying==false)
+          printf("playing note 3\n");
           PATCH_play_note(g_currpatch, 12*_preview_octave, MAX_VELOCITY/2, NULL);
         //SAMPLER_save_sample(plugin, "/tmp/tmp.wav", 0);
       }
diff -ru radium-1.9.1/README radium-master/README
--- radium-1.9.1/README	2012-11-10 19:41:11.000000000 +0100
+++ radium-master/README	2012-11-12 09:08:12.000000000 +0100
@@ -1,7 +1,7 @@
 
 
-Radium V1.9.1
-Released 2012-11-10
+Radium V1.9.2
+Released 2012-11-12
 
 
 
