--- bpmdj-2.9.old/makefile	2005-10-09 16:49:36.000000000 +0200
+++ bpmdj-2.9/makefile		2006-03-11 17:35:04.551889734 +0100
@@ -1,5 +1,6 @@
 VERSION = 2.9
-DESTDIR = /usr/local/
+PREFIX = /usr/local
+DESTDIR = 
 #TIME = /usr/bin/time -f ' '\[%e\]
 #ECHO = echo -n 
 TIME = 
@@ -146,7 +147,7 @@
 
 bpmdj-bin.tgz: install_files bin
 	@echo "[strip]"
-	@strip * 2>/dev/null; exit 0
+	#@strip * 2>/dev/null; exit 0
 	@echo "  [tgz] install"
 	@tar -cz --no-recursion --ignore-failed-read -T install_files -f bpmdj-bin.tgz 2>/dev/null 
 
@@ -190,10 +191,17 @@
 	emacs bpmdj-doc-errors.txt &
 
 install: bpmdj-bin.tgz
-	mkdir -p $(DESTDIR)/bpmdj;
-	cp bpmdj-bin.tgz $(DESTDIR)/bpmdj
-	cd $(DESTDIR)/bpmdj; tar -xzf bpmdj-bin.tgz
-	rm $(DESTDIR)/bpmdj/bpmdj-bin.tgz
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/bpmdj/tmp $(DESTDIR)$(PREFIX)/bin	
+	rm $(DESTDIR)$(PREFIX)/share/bpmdj/tmp/* -rf
+	tar -xzf bpmdj-bin.tgz --directory=$(DESTDIR)$(PREFIX)/share/bpmdj/tmp
+	find $(DESTDIR)$(PREFIX)/share/bpmdj/tmp -perm -700 \
+		-exec $(INSTALL) -m755 {} $(DESTDIR)$(PREFIX)/bin \;
+	find $(DESTDIR)$(PREFIX)/share/bpmdj/tmp -iname '*.pl' \
+		-exec $(INSTALL) -m755 {} $(DESTDIR)$(PREFIX)/bin \;
+	mv $(DESTDIR)$(PREFIX)/share/bpmdj/tmp/sequences/ \
+		$(DESTDIR)$(PREFIX)/share/bpmdj/
+	rm $(DESTDIR)$(PREFIX)/share/bpmdj/tmp/* -rf
+	rmdir $(DESTDIR)$(PREFIX)/share/bpmdj/tmp/
 
 source.tgz-dist: bpmdj-source.tgz
 	@echo "  [tgz] "bpmdj-$(VERSION).source.tgz
