--- daw/mixer/doc/Makefile.old	2010-04-23 19:42:45.000000000 +0100
+++ daw/mixer/doc/Makefile	2010-04-23 19:43:38.000000000 +0100
@@ -17,8 +17,8 @@
 	@ rm -f index.html
 
 install:
-	@ mkdir -p $(DOCUMENT_PATH)/non-mixer
-	@ cp $(OBJS) *.png mup.css ../../COPYING $(DOCUMENT_PATH)/non-mixer
+	@ install -d $(DESTDIR)$(DOCUMENT_PATH)/non-mixer
+	@ install -m 644 $(OBJS) *.png mup.css ../../COPYING $(DESTDIR)$(DOCUMENT_PATH)/non-mixer
 #	@ ln -sf $(PIXMAP_PATH)/logo.png $(DOCUMENT_PATH)
 
 clean:
--- daw/mixer/Makefile.old	2010-04-23 19:36:42.000000000 +0100
+++ daw/mixer/Makefile	2010-04-23 19:38:30.000000000 +0100
@@ -107,15 +107,15 @@
 
 install: all
 	@ echo -n "Installing..."
-	@ install src/mixer $(prefix)/bin/non-mixer
-	@ mkdir -p $(SYSTEM_PATH)/non-mixer
-	@ mkdir -p $(PIXMAP_PATH)/non-mixer
-	@ cp pixmaps/*.png $(PIXMAP_PATH)/non-mixer
+	@ install -D -m 755 src/mixer $(DESTDIR)$(prefix)/bin/non-mixer
+	@ install -d $(DESTDIR)$(SYSTEM_PATH)/non-mixer
+	@ install -d $(DESTDIR)$(PIXMAP_PATH)/non-mixer
+	@ install -m 644 pixmaps/*.png $(DESTDIR)$(PIXMAP_PATH)/non-mixer
 	@ $(MAKE) -s -C doc install
 	@ echo "$(DONE)"
 ifneq ($(USE_DEBUG),yes)
 	@ echo -n "Stripping..."
-	@ strip $(prefix)/bin/non-mixer
+	@ strip $(DESTDIR)$(prefix)/bin/non-mixer
 	@ echo "$(DONE)"
 endif
 
--- daw/timeline/doc/Makefile.old	2010-04-23 19:30:34.000000000 +0100
+++ daw/timeline/doc/Makefile	2010-04-23 19:34:26.000000000 +0100
@@ -17,8 +17,8 @@
 	@ rm -f index.html
 
 install:
-	@ mkdir -p $(DOCUMENT_PATH)/non-daw
-	@ cp $(OBJS) *.png mup.css ../../COPYING $(DOCUMENT_PATH)/non-daw
+	@ install -d $(DESTDIR)$(DOCUMENT_PATH)/non-daw
+	@ install -m 644 $(OBJS) *.png mup.css ../../COPYING $(DESTDIR)$(DOCUMENT_PATH)/non-daw
 #	@ ln -sf $(PIXMAP_PATH)/logo.png $(DOCUMENT_PATH)
 
 clean:
--- daw/timeline/Makefile.old	2010-04-23 18:16:13.000000000 +0100
+++ daw/timeline/Makefile	2010-04-23 18:37:03.000000000 +0100
@@ -119,15 +119,15 @@
 
 install: all
 	@ echo -n "Installing..."
-	@ install src/timeline $(prefix)/bin/non-daw
-	@ mkdir -p $(SYSTEM_PATH)/non-daw
-	@ mkdir -p $(PIXMAP_PATH)/non-daw
-	@ cp pixmaps/*.png $(PIXMAP_PATH)/non-daw
+	@ install -D -m 755 src/timeline $(DESTDIR)$(prefix)/bin/non-daw
+	@ install -d $(DESTDIR)$(SYSTEM_PATH)/non-daw
+	@ install -d $(DESTDIR)$(PIXMAP_PATH)/non-daw
+	@ install -m 644 pixmaps/*.png $(DESTDIR)$(PIXMAP_PATH)/non-daw
 	@ $(MAKE) -s -C doc install
 	@ echo "$(DONE)"
 ifneq ($(USE_DEBUG),yes)
 	@ echo -n "Stripping..."
-	@ strip $(prefix)/bin/non-daw
+	@ strip $(DESTDIR)$(prefix)/bin/non-daw
 	@ echo "$(DONE)"
 endif
 
