--- gst-buzztard-0.5.0/configure.ac.old	2010-10-23 13:49:23.000000000 +0100
+++ gst-buzztard-0.5.0/configure.ac	2010-10-23 14:05:37.000000000 +0100
@@ -229,12 +229,15 @@
 
 
 dnl - Check for buzzmachineloader (bml)
-PKG_CHECK_MODULES(BML, libbml >= 0.5.0 liboil-0.3 >= 0.3.8,
-    have_bml=yes, have_bml=no)
-AC_SUBST(BML_CFLAGS)
-AC_SUBST(BML_LIBS)
-AM_CONDITIONAL(BML_SUPPORT, test "x$have_bml" = "xyes")
-if test "x$have_bml" = "xyes"
+AC_ARG_WITH([bml], AS_HELP_STRING([--without-bml],
+	    [Build without bml library (default: test)]))
+if test "x$with_bml" != "xno" ; then
+  PKG_CHECK_MODULES(BML, libbml >= 0.5.0 liboil-0.3 >= 0.3.8)
+  AC_SUBST(BML_CFLAGS)
+  AC_SUBST(BML_LIBS)
+fi
+AM_CONDITIONAL(BML_SUPPORT, test "x$with_bml" = "xyes")
+if test "x$with_bml" = "xyes"
 then
   BML_DOC_TYPES=
 else
@@ -259,12 +262,15 @@
 
 
 dnl - Check for FluidSynth
-PKG_CHECK_MODULES(FLUIDSYNTH, fluidsynth >= 1.0,
-    have_fluidsynth=yes, have_fluidsynth=no)
-AC_SUBST(FLUIDSYNTH_CFLAGS)
-AC_SUBST(FLUIDSYNTH_LIBS)
-AM_CONDITIONAL(FLUIDSYNTH_SUPPORT, test "x$have_fluidsynth" = "xyes")
-if test "x$have_fluidsynth" = "xyes"
+AC_ARG_WITH([fluidsynth], AS_HELP_STRING([--without-fluidsynth],
+	    [Build without fluidsynth support (default: test)]))
+if test "x$with_fluidsynth" != "xno"; then
+  PKG_CHECK_MODULES(FLUIDSYNTH, fluidsynth >= 1.0)
+  AC_SUBST(FLUIDSYNTH_CFLAGS)
+  AC_SUBST(FLUIDSYNTH_LIBS)
+fi
+AM_CONDITIONAL(FLUIDSYNTH_SUPPORT, test "x$with_fluidsynth" = "xyes")
+if test "x$with_fluidsynth" = "xyes"
 then
   FLUIDSYNTH_DOC_TYPES=
 else
@@ -337,15 +343,15 @@
 	Compiler                   : ${CC}
 
 	Documentation (API)        : ${enable_gtk_doc}
-	Buzzmachine support        : ${have_bml}
-	FluidSynth support         : ${have_fluidsynth}
+	Buzzmachine support        : ${with_bml}
+	FluidSynth support         : ${with_fluidsynth}
 
 	Debug                      : ${enable_debug}
 	Coverage profiling         : ${enable_coverage}
 	Use of deprecated API      : ${set_enable_deprecated}
 "
 
-if test "x${have_bml}" = "xno" -o "x${have_fluidsynth}" = "xno" ; then
+if test "x${with_bml}" = "xno" -o "x${with_fluidsynth}" = "xno" ; then
 echo "
 Some features are not built. If you like to have them built, please check that
 you have the required -devel packages installed and that they can be found in
