From d176b5d168ff7fd016fd83c7fadde843c2e29a45 Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Wed, 26 Sep 2012 16:19:56 +0100
Subject: [PATCH 2/4] remove -O2

---
 SConstruct |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/SConstruct b/SConstruct
index 0798516..b3a4182 100644
--- a/SConstruct
+++ b/SConstruct
@@ -410,7 +410,7 @@ if env['DEBUG']:
     print "Doing a DEBUG build"
     env.MergeFlags( "-DDEBUG -Wall -g" )
 else:
-    env.MergeFlags( "-O2 -DNDEBUG" )
+    env.MergeFlags( "-DNDEBUG" )
 
 if env['PROFILE']:
     print "Doing a PROFILE build"
-- 
1.7.8.6

